---
title: "el_cl_plotting2"
author: "layla"
date: "2025-11-26"
output: html_document
---

```{r}
setwd("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/")
```


### packages ---
```{r message=FALSE, warning=FALSE}
library(readr)
library(magrittr)
library(dplyr)
#library(variancePartition) for R 4.5 only 
library(ggplot2)
library(igraph)
library(ggraph)
library(clusterProfiler)
library(org.Hs.eg.db)
library(enrichplot)
organism = "org.Hs.eg.db"
library(organism, character.only = TRUE)
library(ggrepel)
library(purrr)
library(broom)
library(tibble)
library(stringr)
library(stringdist)
library(RColorBrewer)
library(scales)
library(colorspace)
library(ggmap)
library(forcats)
library(ggridges)
library(corrplot)
library(pheatmap)
library(AnnotationDbi)
library(UpSetR)
library(emmeans)
library(readxl)
library(biomaRt)
library(rrvgo)
```



### age and sex plot ---
 
```{r}
meta <- readRDS("meta_data_13nov25_AFTER_group_fixes.rds")
```

```{r}
meta %>% ggplot(aes(x=age, fill=sex_medical)) + geom_density(alpha=0.8) + theme_bw(base_size = 20) + scale_fill_manual(values = c("#17154FFF","#D95E31FF")) + labs(x = "Age", y="Density", fill="Sex")

#L 8x5
```



### map and EL vs CL plot ---

```{r}
meta <- readRDS("meta_data_13nov25_AFTER_group_fixes.rds")
```

```{r}
meta %>% ggplot(aes(x=Early_urban_PC1, y=urb_score)) + geom_point(color="black") + theme_classic(base_size = 20) + labs(x = "Early Life Urbanicity", y = "Current Life Urbanicity")
#L 6x5
```

```{r}
OA_merged=read.delim('~/Downloads/18apr25_OA_merged_filt.txt')
medianlat <- median(OA_merged$lat, na.rm = TRUE)
medianlong <- median(OA_merged$long, na.rm = TRUE)

pal=c("#5495CFFF", "#F5AF4DFF", "#DB4743FF", "#7C873EFF", "#FEF4D5FF")

register_google('AIzaSyBqRWaHZTtsvaSOIwDdHjmTVPD-cXvaKrE')
s <- "element:geometry%7Ccolor:0xf5f5f5&style=element:labels%7Cvisibility:off&style=element:labels.icon%7Cvisibility:off&style=element:labels.text.fill%7Ccolor:0x616161&style=element:labels.text.stroke%7Ccolor:0xf5f5f5&style=feature:administrative%7Celement:geometry%7Cvisibility:off&style=feature:administrative.country%7Celement:geometry.stroke%7Ccolor:0x000000%7Cvisibility:on&style=feature:administrative.land_parcel%7Cvisibility:off&style=feature:administrative.land_parcel%7Celement:labels.text.fill%7Ccolor:0xbdbdbd&style=feature:administrative.neighborhood%7Cvisibility:off&style=feature:poi%7Cvisibility:off&style=feature:poi%7Celement:geometry%7Ccolor:0xeeeeee&style=feature:poi%7Celement:labels.text.fill%7Ccolor:0x757575&style=feature:poi.park%7Celement:geometry%7Ccolor:0xe5e5e5&style=feature:poi.park%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&style=feature:road%7Cvisibility:off&style=feature:road%7Celement:geometry%7Ccolor:0xffffff&style=feature:road%7Celement:labels.icon%7Cvisibility:off&style=feature:road.arterial%7Celement:labels.text.fill%7Ccolor:0x757575&style=feature:road.highway%7Celement:geometry%7Ccolor:0xdadada&style=feature:road.highway%7Celement:labels.text.fill%7Ccolor:0x616161&style=feature:road.local%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&style=feature:transit%7Cvisibility:off&style=feature:transit.line%7Celement:geometry%7Ccolor:0xe5e5e5&style=feature:transit.station%7Celement:geometry%7Ccolor:0xeeeeee&style=feature:water%7Celement:geometry%7Ccolor:0xc9c9c9&style=feature:water%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&size=480x360"
c(medianlong, medianlat)

map <- get_googlemap(center=c(102,4), zoom = 7, scale = 1, style = s)
m <- ggmap(map)

Current_urb_by_village<- OA_merged %>%
  group_by(village_id) %>%
  summarize(
    mean_lat= mean(lat),
    mean_long=mean(long),
    mean_urb = mean(urb_score, na.rm=T), 
    count = n() )

Current_urb_by_village<-subset(Current_urb_by_village,count>9)
current_urb_filt = Current_urb_by_village[-which(Current_urb_by_village$village_id==11),]
current_urb_plot<- m +
  geom_point(data = current_urb_filt, aes(x = mean_long, y = mean_lat, fill=mean_urb, size=count), color="black", alpha = 0.8, shape=21) +
  labs(x="Longitude", y="Latitude", size="Sample \n  size", fill="    Adult \n urbanicity \n    score") +
  scale_fill_gradient(low = "#5CB9F4", high = pal[3], breaks = c(2.984, 41.729), labels = c("Low", "High")) + 
  scale_size_continuous(range = c(5,7)) + 
  xlim(100.5, 103.5) +
  ylim(2.5, 6) +
  theme_bw(13); current_urb_plot

current_urb_plot<- m +
  geom_point(data = current_urb_filt, aes(x = mean_long, y = mean_lat, fill=mean_urb, size=count), color="black", alpha = 0.8, shape=21) +
  labs(x="Longitude", y="Latitude", size="Sample \n  size", fill="    Adult \n urbanicity \n    score") +
  scale_fill_gradient(low = "#9DC398", high = "#1D2D1A", breaks = c(2.984, 41.729), labels = c("Low", "High")) + 
  scale_size_continuous(range = c(5,7)) + 
  xlim(100.5, 103.5) +
  ylim(2.5, 6) +
  theme_bw(13); current_urb_plot

#L 5x5
```



### tf code ---

```{r}
rm(list = ls())
```

```{r}
tf_targets <- read.delim("~/Desktop/vandy/lea lab/oa/el_cl/figures/trrust_rawdata.human.tsv", header=FALSE, stringsAsFactors=FALSE)
colnames(tf_targets) <- c("TF", "Target", "Mode", "PubMed")

model_out <- readRDS("lmm_grm_evryne_grp_fixed_effect_13nov25.rds")

model_out$gene = rownames(model_out)

model_out$padj_urb = p.adjust(model_out$p_value_urb_scale, method = "fdr")
model_out$padj_early = p.adjust(model_out$p_value_early_scale, method = "fdr")

urb_genes = model_out %>% filter(padj_urb<0.1)
early_genes = model_out %>% filter(padj_early<0.1)

current_urb = urb_genes %>% filter(beta_urb_scale>0) %>% dplyr::select(beta_urb_scale, gene)
current_rural = urb_genes %>% filter(beta_urb_scale<0) %>% dplyr::select(beta_urb_scale, gene)
early_urb = early_genes %>% filter(beta_early_scale>0) %>% dplyr::select(beta_early_scale, gene)
early_rural = early_genes %>% filter(beta_early_scale<0) %>% dplyr::select(beta_early_scale, gene)

tf_list <- split(tf_targets$Target, tf_targets$TF)

#filter to be genes that are significant in any model 
gene_list = c(urb_genes$gene, early_genes$gene)
gene_list=gene_list[!duplicated(gene_list)]

filtered_tf_list <- lapply(tf_list, function(genes) {intersect(genes, gene_list)})

filtered_tf_list <- Filter(length, filtered_tf_list)

# Filter TFs to those that:
# 1. Have >2 targets remaining
# 2. Are themselves also in the gene_list
filtered_tf_list <- filtered_tf_list[names(filtered_tf_list) %in% model_out$gene]
filtered_tf_list <- Filter(function(targets) length(targets) > 4, filtered_tf_list)
```


similarity function to combine TFs
```{r}
# --- 1. jaccard similarity function ---
jaccard_sim <- function(a, b) {
  length(intersect(a, b)) / length(union(a, b))
}

tf_names <- names(filtered_tf_list)

# --- 2. compute similarity matrix ---
sim_mat <- matrix(0, nrow = length(tf_names), ncol = length(tf_names),
                  dimnames = list(tf_names, tf_names))

for (i in seq_along(tf_names)) {
  for (j in seq_along(tf_names)) {
    if (i < j) {
      sim_val <- jaccard_sim(filtered_tf_list[[i]], filtered_tf_list[[j]])
      sim_mat[i, j] <- sim_val
      sim_mat[j, i] <- sim_val
    }
  }
}

# --- 3. convert similarity to distance ---
dist_mat <- as.dist(1 - sim_mat)  # 1 - Jaccard similarity

# --- 4. hierarchical clustering ---
hc <- hclust(dist_mat, method = "average")  # UPGMA

# --- 5. cut tree at Jaccard ≥ 0.7 (distance ≤ 0.3) ---
groups <- cutree(hc, h = 0.7)

# --- 6. collapse TFs within each cluster ---
collapsed_list <- lapply(split(tf_names, groups), function(tf_group) {
  unique(unlist(filtered_tf_list[tf_group]))
})

# give merged names that reflect all TFs in the cluster
names(collapsed_list) <- sapply(split(tf_names, groups), paste, collapse = "_")

```


making sure groups are unique
```{r}
# Extract vectors of genes
g1 <- current_rural$gene
g2 <- current_urb$gene
g3 <- early_rural$gene
g4 <- early_urb$gene

# Combine into a list
gene_lists <- list(current_rural = g1, current_urb = g2, early_rural = g3, early_urb = g4)

# Which genes appear in more than one group?
overlapping_genes <- names(table(unlist(gene_lists)))[table(unlist(gene_lists)) > 1]

combn(names(gene_lists), 2, function(x) {
  overlap <- intersect(gene_lists[[x[1]]], gene_lists[[x[2]]])
  cat("\nOverlap between", x[1], "and", x[2], ":\n")
  print(overlap)
})

all_gene_lists <- list(
  urb = current_urb$gene,
  rural = current_rural$gene,
  early_urb = early_urb$gene,
  early_rural = early_rural$gene)

# Find genes that are in >1 group
gene_counts <- table(unlist(all_gene_lists))
overlapping_genes <- names(gene_counts[gene_counts > 1])
```



```{r}
genes = data.frame(model_out$gene)

genes$current_rural <- ifelse(genes$model_out.gene %in% g1 & !(genes$model_out.gene %in% overlapping_genes), 1, 0)
genes$current_urb <- ifelse(genes$model_out.gene %in% g2 & !(genes$model_out.gene %in% overlapping_genes), 1, 0)
genes$early_rural <- ifelse(genes$model_out.gene %in% g3 & !(genes$model_out.gene %in% overlapping_genes), 1, 0)
genes$early_urb <- ifelse(genes$model_out.gene %in% g4 & !(genes$model_out.gene %in% overlapping_genes), 1, 0)

```



```{r eval=FALSE}
run_tf_enrichment <- function(gene_list, collapsed_list, total_background, exclude_genes = NULL) {
  # Optionally exclude overlapping genes
  if (!is.null(exclude_genes)) {
    gene_vector <- setdiff(gene_list, exclude_genes)
  } else {
    gene_vector <- gene_list
  }

  # Initialize results list
  results <- list()

  # Iterate over all TFs in collapsed_list
  for (tf in names(collapsed_list)) {
    tf_genes <- unique(collapsed_list[[tf]])

    overlap <- length(intersect(gene_vector, tf_genes))
    only_in_list <- length(setdiff(gene_vector, tf_genes))
    only_in_tf <- length(setdiff(tf_genes, gene_vector))
    neither <- total_background - (overlap + only_in_list + only_in_tf)

    # Fisher's exact test
    fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
    fisher_test <- fisher.test(fisher_matrix)

    odds_ratio <- unname(fisher_test$estimate)
    p_value <- unname(fisher_test$p.value)

    # Expected overlap (hypergeometric expectation)
    expected_overlap <- (length(gene_vector) * length(tf_genes)) / total_background

    # Store results
    results[[tf]] <- data.frame(
      TF = tf,
      p.value = p_value,
      odds_ratio = odds_ratio,
      overlap = overlap,
      only_in_list = only_in_list,
      only_in_tf = only_in_tf, 
      neither = neither, 
      total_targets = length(tf_genes),
      expected = expected_overlap,
      stringsAsFactors = FALSE
    )
  }

  # Combine all TF results into one data frame
  result_df <- do.call(rbind, results)

  return(result_df)
}
```


```{r}
current_urb_tf <- run_tf_enrichment(gene_list = current_urb$gene, collapsed_list = collapsed_list, total_background = 9993, exclude_genes = overlapping_genes)

current_rural_tf <- run_tf_enrichment(gene_list = current_rural$gene, collapsed_list = collapsed_list, total_background = 9993, exclude_genes = overlapping_genes)

early_urb_tf <- run_tf_enrichment(gene_list = early_urb$gene, collapsed_list = collapsed_list, total_background = 9993, exclude_genes = overlapping_genes)

early_rural_tf <- run_tf_enrichment(gene_list = early_rural$gene, collapsed_list = collapsed_list, total_background = 9993, exclude_genes = overlapping_genes)
```


```{r}
current_urb_tf$type = "current_urb"
current_rural_tf$type = "current_rural"
early_urb_tf$type = "early_urb"
early_rural_tf$type = "early_rural"

combined_df = rbind(current_urb_tf, current_rural_tf, early_urb_tf, early_rural_tf)
```

```{r}
combined_df_filt = combined_df %>% filter(!odds_ratio=="Inf")
combined_df_filt$adj.p.value = p.adjust(combined_df_filt$p.value, method="fdr")
combined_df_filt = combined_df_filt %>% filter(p.value < 0.1)
combined_df_filt = combined_df_filt %>% mutate(significance = ifelse(adj.p.value < 0.1, "p.adj < 0.1", "p.value < 0.1"))

combined_df_filt <- combined_df_filt %>%
  mutate(TF_type = paste(TF, type, sep = "_")) %>%
  arrange(type, desc(odds_ratio)) %>%
  mutate(TF_type = factor(TF_type, levels = unique(TF_type)))

combined_df_filt$TF_label <- combined_df_filt$TF 
```

```{r}
top5_df <- combined_df_filt %>%
  group_by(type) %>%
  arrange(adj.p.value, .by_group = TRUE) %>%
  slice_head(n = 5) %>% ungroup()

ggplot(top5_df, aes(x = TF_type, y = log2(odds_ratio), fill = type)) +
  geom_col(color = "black") +
  scale_x_discrete(labels = top5_df$TF_label) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 20),
    axis.text.y = element_text(size = 20), legend.text = element_text(size = 20), text = element_text(size = 20)) +
  scale_fill_manual(values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  labs(x = "TF", y = "log2(Odds Ratio)")
```

chart showing TF sharing 
```{r}
top5_df <- combined_df_filt %>%
  group_by(type) %>%
  arrange(adj.p.value, .by_group = TRUE) %>%
  slice_head(n = 5) %>%
  ungroup()

top5_df <- top5_df %>%
  mutate(TF = factor(TF, levels = unique(TF)))

ggplot(top5_df, aes(x = type, y = TF, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(
    values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  theme_bw() +
  labs(title = "Top 5 TFs per Category", x = "Type", y = "Transcription Factor") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), panel.grid = element_blank()) 

#L 5x5
```

alpha by odds ratio 
```{r}
ggplot(top5_df, aes(x = type, y = TF, fill = type, alpha = log2(odds_ratio))) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(
    values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  scale_alpha_continuous(range = c(0.6, 1), name = "log2(odds ratio)") +
  theme_bw(base_size = 20) +
  labs(title = "Top 5 TFs per Category", x = "Type", y = "Transcription Factor") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
#8x8 L
```

```{r}
ggplot(top5_df, aes(x = type, y = TF, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(
    values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  theme_bw(base_size = 20) +
  labs(title = "Top 5 TFs per Category", x = "Type", y = "Transcription Factor") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```


looking at nfkb1_rela targets 
```{r}
# Pull NFKB1_RELA target genes from your collapsed_list
nfkb_targets <- unique(collapsed_list[["NFKB1_RELA"]])

# For each category, find which genes overlap
nfkb_df <- bind_rows(
  data.frame(gene = intersect(current_urb$gene, nfkb_targets),   type = "current_urb"),
  data.frame(gene = intersect(current_rural$gene, nfkb_targets), type = "current_rural"),
  data.frame(gene = intersect(early_urb$gene, nfkb_targets),     type = "early_urb"),
  data.frame(gene = intersect(early_rural$gene, nfkb_targets),   type = "early_rural"))

nfkb_rela_targets <- tf_targets %>%
  filter(TF %in% c("NFKB1", "RELA"))

nfkb_annot <- nfkb_df %>%
    inner_join(nfkb_rela_targets, by = c("gene" = "Target")) %>%
    group_by(gene, type) %>%
    summarise(
        TF = paste(unique(TF), collapse = "+"),
        direction = paste(unique(Mode), collapse = "+"),
        .groups = "drop")
```



addinging heirarchial clustering 
```{r}
nfkb_df <- bind_rows(
  current_urb  %>% filter(gene %in% nfkb_targets)  %>% transmute(gene, type = "current_urb",  beta = beta_urb_scale),
  current_rural %>% filter(gene %in% nfkb_targets) %>% transmute(gene, type = "current_rural", beta = beta_urb_scale),
  early_urb   %>% filter(gene %in% nfkb_targets)   %>% transmute(gene, type = "early_urb",   beta = beta_early_scale),
  early_rural %>% filter(gene %in% nfkb_targets)   %>% transmute(gene, type = "early_rural", beta = beta_early_scale))
```

```{r}
mat <- nfkb_df %>%
  tidyr::pivot_wider(names_from = type, values_from = beta, values_fill = 0) %>%
  column_to_rownames("gene")

# cluster rows (genes) and columns (categories)
row_order <- hclust(dist(mat))$order
col_order <- hclust(dist(t(mat)))$order

# apply ordering
nfkb_df$gene <- factor(nfkb_df$gene, levels = rownames(mat)[row_order])
nfkb_df$type <- factor(nfkb_df$type, levels = colnames(mat)[col_order])

# plot
ggplot(nfkb_df, aes(x = type, y = gene, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2")) + 
  theme_bw(base_size = 20) +
  labs(title = "NFKB1_RELA Targets", x = "Type", y = "Gene") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
```



 
### urb rural enrich plot ---

```{r}
rm(list = ls())
organism = "org.Hs.eg.db"
```

```{r}
set.seed(24)
```

read in model results 
```{r}
model_out <- readRDS("lmm_grm_evryne_grp_fixed_effect_13nov25.rds")

model_out$gene = rownames(model_out)

model_out$padj_urb = p.adjust(model_out$p_value_urb_scale, method = "fdr")
model_out$padj_early = p.adjust(model_out$p_value_early_scale, method = "fdr")

urb_genes = model_out %>% filter(padj_urb<0.1)
early_genes = model_out %>% filter(padj_early<0.1)

current_urb = urb_genes %>% filter(beta_urb_scale>0) %>% dplyr::select(beta_urb_scale, gene)
current_rural = urb_genes %>% filter(beta_urb_scale<0) %>% dplyr::select(beta_urb_scale, gene)
early_urb = early_genes %>% filter(beta_early_scale>0) %>% dplyr::select(beta_early_scale, gene)
early_rural = early_genes %>% filter(beta_early_scale<0) %>% dplyr::select(beta_early_scale, gene)

```


CL urban, background (all genes), test (significant CL urban genes)
```{r}
test= subset(model_out, model_out$padj_urb<0.1 & model_out$beta_urb_scale>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$qvalue < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
```




CL rural, background (all genes), test (significant CL rural genes)
```{r}
test= subset(model_out, model_out$padj_urb<0.1 & model_out$beta_urb_scale<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$pvalue < 0.05, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
```



```{r}
# Add a column to each df to mark Urban vs Rural
urban_df <- data.frame(filtered_urban %>% mutate(Group = "Urban"))
rural_df <- data.frame(filtered_rural %>% mutate(Group = "Rural"))

# Take top 5 from each based on adjusted p-value
urban_top5 <- urban_df %>% arrange(p.adjust) %>% slice_head(n = 5)
rural_top5 <- rural_df %>% arrange(p.adjust) %>% slice_head(n = 5)

df_top <- bind_rows(urban_top5, rural_top5) %>%
  mutate(Description_clean = Description)

df_top <- df_top %>%
  group_by(Group) %>%
  arrange(desc(FoldEnrichment), .by_group = TRUE) %>%
  mutate(ordering = row_number()) %>%
  ungroup()


manual_order <- c(
  # Rural top 5
  "enzyme-linked receptor protein signaling pathway",
  "regulation of small GTPase mediated signal transduction",
  "adenylate cyclase-modulating G protein-coupled receptor signaling pathway",
  "adenylate cyclase-activating G protein-coupled receptor signaling pathway",
  "lens development in camera-type eye",
  
  # Urban top 5
  "organic acid catabolic process",
  "carboxylic acid catabolic process",
  "small molecule catabolic process",
  "monocarboxylic acid metabolic process",
  "carboxylic acid metabolic process")

# Apply to factor
df_top <- df_top %>%
  mutate(Description_clean = factor(Description_clean, levels = manual_order))
```

```{r}
ggplot(df_top, aes(x = Description_clean, y = 0)) +
  geom_point(aes(
    size = Count,
    color = Group,
    #alpha = FoldEnrichment
    )) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "darkolivegreen", "Urban" = "darkolivegreen3")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group",
    alpha = "Fold Enrichment")

#L 8x7
```


```{r}
urb_simMatrix <- calculateSimMatrix(urban_df$ID, org.Hs.eg.db, ont="BP")
urb_id_scores <- setNames(-log10(urban_df$p.adjust), urban_df$ID)
urb_reduced <- reduceSimMatrix(urb_simMatrix,
                     urb_id_scores,
                     threshold = 0.7,   # higher = bigger clusters
                     orgdb = org.Hs.eg.db)
#saveRDS(urb_reduced, file = "~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/CL_urb_reduced_26nov25.rds")
```

```{r}
rural_simMatrix <- calculateSimMatrix(rural_df$ID, org.Hs.eg.db, ont="BP")
rural_id_scores <- setNames(-log10(rural_df$p.adjust), rural_df$ID)
rural_reduced <- reduceSimMatrix(rural_simMatrix,
                     rural_id_scores,
                     threshold = 0.7,   # higher = bigger clusters
                     orgdb = org.Hs.eg.db)
#saveRDS(rural_reduced, file = "~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/CL_rural_reduced_26nov25.rds")
```

```{r}
urb_reduced <- readRDS("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/CL_urb_reduced_26nov25.rds")
rural_reduced <- readRDS("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/CL_rural_reduced_26nov25.rds")

urban_top_counts <- urb_reduced %>% count(parentTerm, name = "n_terms") %>% arrange(desc(n_terms)) %>% slice_max(n_terms, n = 5)
urban_top_counts$group = "Urban"

rural_top_counts <- rural_reduced %>%  count(parentTerm, name = "n_terms") %>% slice_max(n_terms, n = 5)
rural_top_counts$group = "Rural"

df_top = rbind(urban_top_counts, rural_top_counts)

manual_order <- c(
  # Rural top 5
  "T cell costimulation",
  "coronary vasculature morphogenesis",
  "regulation of cellular response to growth factor stimulus",
  "centriole-centriole cohesion",
  "negative regulation of protein-containing complex assembly",
  "positive regulation of cell adhesion", 
  "regulation of activated T cell proliferation",
  
  # Urban top 5
  "tumor necrosis factor production", 
  "response to bacterial lipoprotein",
  "endothelial cell migration",
  "diol metabolic process",
  "nucleoside phosphate metabolic process")

# Apply to factor
df_top <- df_top %>%
  mutate(parentTerm = factor(parentTerm, levels = manual_order))


ggplot(df_top, aes(x = parentTerm, y = 0)) +
  geom_point(aes(size = n_terms, color = group)) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "darkolivegreen", "Urban" = "darkolivegreen3")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group")
```


----------------------------------------------------------------------------------------------------------------------------



EL urban, background (all genes), test (significant EL urban genes)
```{r}
test= subset(model_out, model_out$padj_early<0.1 & model_out$beta_early_scale>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$qvalue < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
```

EL rural, background (all genes), test (significant EL rural genes)
```{r}
test= subset(model_out, model_out$padj_early<0.1 & model_out$beta_early_scale<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$qvalue < 0.1, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
```

```{r}
# Add a column to each df to mark Urban vs Rural
urban_df <- data.frame(filtered_urban %>% mutate(Group = "Urban"))
rural_df <- data.frame(filtered_rural %>% mutate(Group = "Rural"))

# Take top 5 from each based on adjusted p-value
urban_top5 <- urban_df %>% arrange(p.adjust) %>% slice_head(n = 5)
rural_top5 <- rural_df %>% arrange(p.adjust) %>% slice_head(n = 5)

df_top <- bind_rows(urban_top5, rural_top5) %>%
  mutate(Description_clean = Description)

df_top <- df_top %>%
  group_by(Group) %>%
  arrange(desc(FoldEnrichment), .by_group = TRUE) %>%
  mutate(ordering = row_number()) %>%
  ungroup()

manual_order <- c(
  # Rural top 5
  "regulation of immune effector process",
  "regulation of body fluid levels",
  "hemostasis",
  "platelet aggregation",
  "exocytic process",
  
  # Urban top 5
  "chaperone-mediated autophagy",
  "'de novo' post-translational protein folding",
  "response to unfolded protein",
  "response to topologically incorrect protein",
  "cellular response to lipid")

# Apply to factor
df_top <- df_top %>%
  mutate(Description_clean = factor(Description_clean, levels = manual_order))
```

```{r}
ggplot(df_top, aes(x = Description_clean, y = 0)) +
  geom_point(aes(
    size = Count,
    color = Group,
    #alpha = FoldEnrichment
    )) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "dodgerblue3", "Urban" = "skyblue2")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group",
    alpha = "Fold Enrichment")

#L 6x5
```

```{r}
urb_simMatrix <- calculateSimMatrix(urban_df$ID, org.Hs.eg.db, ont="BP")
urb_id_scores <- setNames(-log10(urban_df$p.adjust), urban_df$ID)
urb_reduced <- reduceSimMatrix(urb_simMatrix,
                     urb_id_scores,
                     threshold = 0.7,   # higher = bigger clusters
                     orgdb = org.Hs.eg.db)
#saveRDS(urb_reduced, file = "~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/EL_urb_reduced_26nov25.rds")
```

```{r}
rural_simMatrix <- calculateSimMatrix(rural_df$ID, org.Hs.eg.db, ont="BP")
rural_id_scores <- setNames(-log10(rural_df$p.adjust), rural_df$ID)
rural_reduced <- reduceSimMatrix(rural_simMatrix,
                     rural_id_scores,
                     threshold = 0.7,   # higher = bigger clusters
                     orgdb = org.Hs.eg.db)
#saveRDS(rural_reduced, file = "~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/EL_rural_reduced_26nov25.rds")
```

```{r}
urb_reduced <- readRDS("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/EL_urb_reduced_26nov25.rds")
rural_reduced <- readRDS("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/EL_rural_reduced_26nov25.rds")

urban_top_counts <- urb_reduced %>% count(parentTerm, name = "n_terms") %>% arrange(desc(n_terms)) %>% slice_max(n_terms, n = 5)
urban_top_counts$group = "Urban"

rural_top_counts <- rural_reduced %>% count(parentTerm, name = "n_terms") %>% arrange(desc(n_terms)) %>% slice_max(n_terms, n = 5)
rural_top_counts$group = "Rural"

df_top = rbind(urban_top_counts, rural_top_counts)

manual_order <- c(
  # Rural top 5
  "interleukin-6 production",
  "regulation of B cell differentiation",
  "regulation of body fluid levels",
  "anatomical structure homeostasis",
  "monoamine transport",
  "platelet aggregation", 
  "egulation of immune effector process",
  
  # Urban top 5
  "regulation of protein import into nucleus",
  "negative regulation of cytokine production",
  "leukocyte chemotaxis",
  "'de novo' post-translational protein folding",
  "CD4-positive, alpha-beta T cell differentiation involved in immune response")

# Apply to factor
df_top <- df_top %>%
  mutate(parentTerm = factor(parentTerm, levels = manual_order))


ggplot(df_top, aes(x = parentTerm, y = 0)) +
  geom_point(aes(size = n_terms, color = group)) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "dodgerblue3", "Urban" = "skyblue2")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group")
```



### gene expression volcano plots ---

```{r}
rm(list = ls())
```

read in model results 
```{r}
model_out <- readRDS("lmm_grm_evryne_grp_fixed_effect_13nov25.rds")

model_out$gene = rownames(model_out)

model_out$padj_urb = p.adjust(model_out$p_value_urb_scale, method = "fdr")
model_out$padj_early = p.adjust(model_out$p_value_early_scale, method = "fdr")

urb_genes = model_out %>% filter(padj_urb<0.1)
early_genes = model_out %>% filter(padj_early<0.1)

model_out$sig_current = ifelse(model_out$padj_urb<0.1, 1, 0)
model_out$sig_early = ifelse(model_out$padj_early<0.1, 1, 0)
```


```{r}
top_genes <- model_out[model_out$sig_current == 1, ]
top_genes <- top_genes[order(top_genes$p_value_urb_scale, -abs(top_genes$beta_urb_scale)), ][1:10, ]

model_out %>% ggplot(aes(x=beta_urb_scale, y=-log10(p_value_urb_scale))) + geom_point(aes(color = as.factor(sig_current), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "darkgreen"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Current Lifestyle Genes") + ylim(0,10)

#L 5x5
```

```{r}
top_genes_filt = model_out %>% filter(gene %in% c("PDK4", "LCN10"))

model_out %>% ggplot(aes(x=beta_urb_scale, y=-log10(p_value_urb_scale))) + geom_point(aes(color = as.factor(sig_current), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "darkgreen"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes_filt, aes(label = gene), size = 6, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Current Lifestyle Genes") + ylim(0,10) + xlim(-0.5, 0.5)

#L 5x5
```


```{r}
top_genes <- model_out[model_out$sig_early == 1, ]
top_genes <- top_genes[order(top_genes$p_value_early_scale, -abs(top_genes$beta_early_scale)), ][1:10, ]

model_out %>% ggplot(aes(x=beta_early_scale, y=-log10(p_value_early_scale))) + geom_point(aes(color = as.factor(sig_early), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "steelblue"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Early Lifestyle Genes") + ylim(0,10)

#L 5x5
```

```{r}
top_genes_filt = model_out %>% filter(gene %in% c("CXCL1", "CPT2"))

model_out %>% ggplot(aes(x=beta_early_scale, y=-log10(p_value_early_scale))) + geom_point(aes(color = as.factor(sig_early), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "steelblue"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes_filt, aes(label = gene), size = 6, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Early Lifestyle Genes") + ylim(0,10) + xlim(-0.5, 0.5)

#L 5x5
```


```{r}
gene_list = c(urb_genes$gene, early_genes$gene)
gene_list=gene_list[!duplicated(gene_list)]

plot_data <- model_out %>%
  dplyr::select(beta_urb_scale, beta_early_scale) %>%
  tidyr::pivot_longer(cols = everything(),names_to = "variable", values_to = "value")

plot_data %>% filter(abs(value) <0.10) %>% ggplot(aes(x = abs(value), fill = variable)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Gene Expression Effect Sizes", x = "abs(Effect Size)", y = "Density") + theme(legend.position = "none")
#L 10x5


plot_data <- model_out %>% filter(gene %in% gene_list) %>%
  dplyr::select(beta_urb_scale, beta_early_scale) %>%
  tidyr::pivot_longer(cols = everything(),names_to = "variable", values_to = "value")

plot_data %>% filter(abs(value) <0.15) %>% ggplot(aes(x = abs(value), fill = variable)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
#L 5x4
```


```{r}
plot_data %>% ggplot(aes(x= variable, y = abs(value), fill = variable)) + geom_violin(alpha=0.8)+ geom_boxplot(width=0.08)  + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank())  + ggtitle("Gene Expression")

#5x4 p
```

### cytokines ---

```{r}
rm(list = ls())
```

large metadata was made 18apr25 using the el score code
```{r message=FALSE, warning=FALSE}
df_filtered=read.delim('~/Desktop/vandy/lea lab/oa/plasma/18apr25_OA_merged.txt')
biomarkers <- read_csv("~/Desktop/vandy/lea lab/oa/plasma/biomarkers_plasma_1.csv")
```

average repeated samples
```{r warning=FALSE}
columns_to_average <- names(biomarkers)[7:23]
bio_avg <- biomarkers %>%
  group_by(Sample.ID) %>%
  summarise(across(all_of(columns_to_average), mean), .groups = "drop")
#ignore the warnings it's just mad about taking the average of an NA 
```


```{r}
meta_small = df_filtered %>% filter(tid %in% bio_avg$Sample.ID)
bio_merge = merge(df_filtered, bio_avg, by.x = "tid", by.y = "Sample.ID")
```

filter out under 18 individuals
```{r}
meta_small %<>% filter(!age<18)
bio_merge %<>% filter(!age<18)
bio_avg %<>% filter(Sample.ID %in% bio_merge$tid)
```

remove cols with no data or don't need
```{r}
bio_avg %<>% dplyr::select(-CTXI..ng.ml., -Final.Osteocalcin..pg.ml., -PLT_leptin_LBP, -Sample_leptin_LBP)
bio_merge %<>% dplyr::select(-CTXI..ng.ml., -Final.Osteocalcin..pg.ml., -PLT_leptin_LBP, -Sample_leptin_LBP)
```


```{r}
cytokine_cols <- names(bio_avg)[2:14]
# 1. Log-transform (with small constant to avoid log(0))
bio_merge[cytokine_cols] <- log2(bio_merge[cytokine_cols] + 0.1)

# 2. Z-score scale (center and scale each column)
bio_merge[cytokine_cols] <- scale(bio_merge[cytokine_cols])
```

modeling urb_score and early life together 
```{r}
safe_model <- safely(function(colname) {
  formula_str <- paste0(
    colname, " ~ scale(urb_score) + scale(Early_urban_PC1) + scale(age) + sex_medical"
  )
  f <- as.formula(formula_str)
  model <- lm(f, data = bio_merge)

  # Get tidy model summary and number of observations
  list(
    tidy = broom::tidy(model),    # this includes Estimate, Std. Error, t value, Pr(>|t|)
    n = nobs(model)
  )
})


# Apply to named list so names are preserved
model_results <- map(set_names(cytokine_cols, cytokine_cols), safe_model)

# Separate success/failure
successful <- keep(model_results, ~ is.null(.x$error))
failed     <- purrr::discard(model_results, ~ is.null(.x$error))

# Extract and combine model summaries
model_summary_df <- imap_dfr(successful, function(res, colname) {
  res$result$tidy %>%
    mutate(response = colname, n = res$result$n)
})

model_summary_df$p_adj = p.adjust(model_summary_df$p.value, method='fdr')

model_summary_df$sig = ifelse(model_summary_df$p_adj<0.1, "sig", "ns")

model_summary_df <- model_summary_df %>%
  mutate(significance = case_when(
    p_adj < 0.1 ~ "FDR < 0.1",
    p.value < 0.1 ~ "p value < 0.1",
    TRUE ~ "ns"
  ))
model_summary_df$direction <- ifelse(model_summary_df$response %in% c("Final.IL.10..pg.ml.", "Final.IL.13..pg.ml.", "Final.IL.4..pg.ml."), "anti", "pro")
```



```{r}
manual_order <- c(
  "Final.Leptin..pg.ml.",
  "CRP..ng.ml.",
  "Final.IL.8..pg.ml.",
  "Final.LBP..pg.ml.",
  "Final.IL.2..pg.ml.",
  "Final.IL.6..pg.ml.",
  "Final.IL.1beta..pg.ml.",
  "Final.IL12p70..pg.ml.",
  "Final.TNF.alpha..pg.ml.",
  "Final.IFN.gamma..pg.ml.",
  "Final.IL.10..pg.ml.",
  "Final.IL.13..pg.ml.", 
  "Final.IL.4..pg.ml.")

model_summary_df$response <- factor(model_summary_df$response, levels = rev(manual_order))
```



```{r}
model_summary_df %>%
  filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>%
  ggplot(aes(x = estimate, y = response, 
             fill = term, alpha = significance)) +
  geom_col(position = position_dodge(width = 0.7), width = 0.8) +
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_fill_manual(values = c("scale(urb_score)" = "darkgreen", "scale(Early_urban_PC1)" = "steelblue")) +
  scale_alpha_manual(values = c("FDR < 0.1" = 1, "p value < 0.1" = 0.8, "ns" = 0.3)) +
  labs(x = "Beta Estimate", y = "", fill = "Term", alpha = "Significance") +
  theme_minimal(base_size = 20)


model_summary_df %>%
  filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>%
  ggplot(aes(x = estimate, y = response, fill = term)) +
  geom_col(position = position_dodge(width = 0.7), width = 0.8) +
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_fill_manual(values = c("scale(urb_score)" = "darkgreen", "scale(Early_urban_PC1)" = "steelblue")) +
  labs(x = "Beta Estimate", y = "", fill = "Term") +
  theme_minimal(base_size = 20)

#L 15x7
```


```{r}

model_summary_df %>% filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>% ggplot(aes(x = abs(estimate), fill = term)) + geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Cytokine Effect Sizes", x = "abs(Effect Size)", y = "Density")  + theme(legend.position = "none")
#L 10x5


model_summary_df %>% filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>% filter(!significance=="ns")  %>% ggplot(aes(x = abs(estimate), fill = term)) + geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
#L 5x4
```


```{r}
model_summary_df %>% filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>% filter(!significance=="ns")  %>% ggplot(aes(x = term, y=abs(estimate),fill = term)) + geom_violin(alpha=0.8)+ geom_boxplot(width=0.08)  + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank())  + ggtitle("Cytokine")

#5x4 p
```


```{r}
bio_merge  %>% ggplot(aes(x=urb_score, y=CRP..ng.ml.)) + geom_point(color="darkgreen") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Current Lifestyle Score", y = "CRP")

#5x5 L
```

```{r}
bio_merge  %>% ggplot(aes(x=Early_urban_PC1, y=CRP..ng.ml.)) + geom_point(color="steelblue") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Early Lifestyle Score", y = "CRP")

#5x5 L
```

### cell type proportions modeling ---

```{r}
rm(list = ls())
```

```{r}
proportions = readRDS('out_parallel_9oct25.rds')
colnames(proportions)[1] = "sample_name"

info <- readRDS("meta_final.rds")
info %<>% dplyr::select(urb_score.y, Early_urban_PC1, age_scale, sex_medical, sample_name)

info$urb_score.y = as.numeric(scale(info$urb_score.y))
info$Early_urban_PC1 = as.numeric(scale(info$Early_urban_PC1))
info$sex_medical = ifelse(info$sex_medical=="female", 0, 1)

df = left_join(info, proportions, by="sample_name")
```


model and store results 
```{r}
cell_columns = colnames(df)[6:22]

model_results <- vector("list", length(cell_columns))

for (i in seq_along(cell_columns)) {
  cell_col <- cell_columns[i]
  model_results[[i]] <- lm(as.formula(paste0("`", cell_col, "` ~ urb_score.y + Early_urban_PC1 + age_scale + sex_medical")), data = df)
}

names(model_results) <- cell_columns
```

extract coefficients 
```{r}
coeff_df <- bind_rows(
  lapply(names(model_results), function(name) {
    broom::tidy(model_results[[name]]) %>%
      mutate(cell_type = name)
  }),
  .id = "model"
)

coeff_df$p.adj = p.adjust(coeff_df$p.value, method = "fdr")
```

make plotting df  
```{r}
plot_df <- coeff_df %>%
  filter(term != "(Intercept)") %>%              # remove intercept
  mutate(significance = ifelse(p.adj < 0.1, "p.adj < 0.1", "n.s."))  # add shading category
```

make cell names more intuitive 
```{r}
cell_type_map <- c(
  "Age.associated.B.cells"       = "Age-Associated B",
  "CD16..NK.cells"               = "CD16+ NK",
  "Classical.monocytes"          = "Classical Monocytes",
  "DC2"                          = "DC2",
  "MAIT.cells"                   = "MAIT",
  "Megakaryocytes.platelets"     = "Megakaryocytes/Platelets",
  "Memory.B.cells"               = "Memory B",
  "Naive.B.cells"                = "Naive B",
  "Non.classical.monocytes"      = "Non-Classical Monocytes",
  "Plasma.cells"                 = "Plasma",
  "Regulatory.T.cells"           = "Treg",
  "Tcm.Naive.cytotoxic.T.cells"  = "CD8+ Naive Cytotoxic T",
  "Tcm.Naive.helper.T.cells"     = "CD4+ Naive Helper T",
  "Tem.Effector.helper.T.cells"  = "CD4+ Effector Helper T",
  "Tem.Temra.cytotoxic.T.cells"  = "CD8+ Temra Cytotoxic T",
  "Tem.Trm.cytotoxic.T.cells"    = "CD8+ Memory Cytotoxic T",
  "pDC"                          = "pDC"
)

plot_df <- plot_df %>%
  mutate(cell_type = recode(cell_type, !!!cell_type_map))
```

make broader groups 
```{r}
plot_df <- plot_df %>%
  mutate(cell_group = case_when(
    cell_type %in% c("Treg", "CD8+ Naive Cytotoxic T", "CD4+ Naive Helper T",
                     "CD4+ Effector Helper T", "CD8+ Temra Cytotoxic T", 
                     "CD8+ Memory Cytotoxic T", "MAIT") ~ "T cells",
    
    cell_type %in% c("Classical Monocytes", "Non-Classical Monocytes") ~ "Monocytes",
    
    cell_type %in% c("CD16+ NK") ~ "ILC",
    
    cell_type %in% c("Memory B", "Naive B", "Age-Associated B") ~ "B cells",
    
    cell_type %in% c("Plasma") ~ "Plasma",
    
    cell_type %in% c("Megakaryocytes/Platelets") ~ "Platelets",
    
    cell_type %in% c("DC2", "pDC") ~ "DC",
    
    TRUE ~ "Other"  # default if any unmatched
  ))
```

reorder cell_type factor for plotting
```{r}
cell_order <- plot_df %>%
  filter(term == "urb_score.y") %>%
  arrange(cell_group, desc(estimate)) %>%
  pull(cell_type)

plot_df_filt <- plot_df %>%
  filter(term %in% c("urb_score.y", "Early_urban_PC1")) %>%
  mutate(cell_type = factor(cell_type, levels = cell_order),
         cell_group = factor(cell_group, levels = c("T cells", "ILC", "Monocytes", "B cells", "Plasma", "Platelets", "DC")))
```


```{r}
plot_df_filt %>% ggplot(aes(x = cell_type, y = estimate, fill = term, alpha = significance)) +
  geom_col(position = position_dodge(width = 0.8), width = 0.7) +
  scale_alpha_manual(values = c("p.adj < 0.1" = 1, "n.s." = 0.4)) +
  scale_fill_manual(values = c("steelblue", "darkgreen")) +
  labs(x = "Cell Type", y = "Beta Estimate", fill = "Model Term", alpha = "Significance",
       title = "Model Coefficients Across Cell Types") +
  theme_minimal(base_size = 30) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        legend.position = "right") + coord_flip()

#L 15x10
```


```{r}
plot_df_filt %>% ggplot(aes(x = abs(estimate), fill = term)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Cell Type Effect Sizes", x = "abs(Effect Size)", y = "Density") + theme(legend.position = "none")
#L 10x5


plot_df_filt %>% filter(p.adj<0.1) %>% ggplot(aes(x = abs(estimate), fill = term)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
#L 5x4
```

```{r}
plot_df_filt %>% filter(p.adj<0.1) %>% ggplot(aes(x= term, y = abs(estimate), fill = term)) + geom_violin(alpha=0.8)+ geom_boxplot(width=0.08)  + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank())  + ggtitle("Cell Type")

#5x4 p
```

```{r}
df  %>% ggplot(aes(x=urb_score.y, y=Tem.Trm.cytotoxic.T.cells)) + geom_point(color="darkgreen") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Current Lifestyle Score", y = "CD8+ Memory Cytotoxic T")
#L 5x5 
```


```{r}
df  %>% ggplot(aes(x=Early_urban_PC1, y=Tem.Trm.cytotoxic.T.cells)) + geom_point(color="steelblue") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Early Lifestyle Score", y = "CD8+ Memory Cytotoxic T")
#L 5x5 
```



```{r}
proportions_filt = proportions %>% filter(sample_name %in% df$sample_name) 
colnames(proportions_filt) <- recode(colnames(proportions_filt), !!!cell_type_map)
proportions_long <- proportions_filt %>% tidyr::pivot_longer(cols = -sample_name, names_to = "cell_type", values_to = "proportion")
cell_group_levels <- c("T cells", "ILC", "Monocytes", "B cells", "Plasma", "Platelets", "DC")
cell_type_order <- c("Treg", "CD8+ Naive Cytotoxic T", "CD4+ Naive Helper T", "CD4+ Effector Helper T",
                     "CD8+ Temra Cytotoxic T", "CD8+ Memory Cytotoxic T", "MAIT",
                     "CD16+ NK",
                     "Classical Monocytes", "Non-Classical Monocytes",
                     "Memory B", "Naive B", "Age-Associated B",
                     "Plasma",
                     "Megakaryocytes/Platelets",
                     "DC2", "pDC")

proportions_long <- proportions_long %>%
  mutate(cell_type = factor(cell_type, levels = cell_type_order))

cell_group_map <- c(
  "Treg" = "T cells", "CD8+ Naive Cytotoxic T" = "T cells", "CD4+ Naive Helper T" = "T cells",
  "CD4+ Effector Helper T" = "T cells", "CD8+ Temra Cytotoxic T" = "T cells", "CD8+ Memory Cytotoxic T" = "T cells",
  "MAIT" = "T cells",
  "CD16+ NK" = "ILC",
  "Classical Monocytes" = "Monocytes", "Non-Classical Monocytes" = "Monocytes",
  "Memory B" = "B cells", "Naive B" = "B cells", "Age-Associated B" = "B cells",
  "Plasma" = "Plasma",
  "Megakaryocytes/Platelets" = "Platelets",
  "DC2" = "DC", "pDC" = "DC")

proportions_long <- proportions_long %>%
  mutate(cell_group = recode(cell_type, !!!cell_group_map))
```

```{r}
proportions_long <- proportions_long %>%
  mutate(cell_type = forcats::fct_rev(cell_type))

my_palette <- c(
  "#003464FF", #T cells
  "#F38630FF", #ILC
  "#018571",  #monocyte
  "#4F3855FF", #b cell
  "#F8D564FF", #plasma
  "#2C5724FF", #platelets
  "#9E3D22"  #DC
)

proportions_long %>% ggplot(aes(x = proportion, y = cell_type, fill = cell_group)) +
  geom_density_ridges(scale = 1, alpha = 1, color = "white") +
  labs(x = "Proportion", y = "Cell Type", title = "Distribution of Cell Type Proportions", fill = "Cell Group") +
  theme_minimal(base_size = 16) + xlim(0, 0.15) + scale_fill_manual(values = my_palette)
#L 7.5x5
```


```{r}
proportions_long <- proportions_long %>%
  mutate(cell_type = forcats::fct_rev(cell_type))

proportions_long %>% ggplot(aes(x=sample_name, y=proportion, fill = cell_type)) + geom_bar(stat = "identity", alpha=1) + theme_minimal(base_size = 16) + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + labs(x = "Sample", y = "Proportion", fill = "Cell Type")

#L 7.5x5
```


```{r}
correlation_matrix <- cor(proportions[, -1])  # exclude sample_name column
colnames(correlation_matrix) <- cell_type_map[colnames(correlation_matrix)]
rownames(correlation_matrix) <- cell_type_map[rownames(correlation_matrix)]
```

```{r}
corrplot(correlation_matrix, method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", number.cex = 0.5, mar = c(0, 0, 1, 0), title = "Correlations between Cell Type Proportions in Orang Asli")
```



### umap ---

```{r}
umap_df=readRDS("umap_plot_df_8oct25.rds")

#remove cell types with only 1 cell 
umap_df %<>% filter(!cell_type_high %in% c("Double-positive thymocytes", "Early MK", "Epithelial cells"))

cell_group_map <- c(
  "Regulatory T cells" = "T cells", "Tcm/Naive cytotoxic T cells" = "T cells", "Tcm/Naive helper T cells" = "T cells",
  "Tem/Effector helper T cells" = "T cells", "Tem/Temra cytotoxic T cells" = "T cells", "Tem/Trm cytotoxic T cells" = "T cells",
  "MAIT cells" = "T cells",
  "CD16+ NK cells" = "ILC",
  "Classical monocytes" = "Monocytes", "Non-classical monocytes" = "Monocytes", "Mono-mac" = "Monocytes", "Macrophages" = "Monocytes",
  "Memory B cells" = "B cells", "Naive B cells" = "B cells", "Age-associated B cells" = "B cells",
  "Plasma cells" = "Plasma",
  "Megakaryocytes/platelets" = "Platelets",
  "DC2" = "DC", "pDC" = "DC")

umap_df <- umap_df %>%
  mutate(cell_group = recode(cell_type_low, !!!cell_group_map))
```

color palettes
```{r}

#had to change order to go with order of cell groups

my_palette <- c(
  "#4F3855FF", #b cell
  "#F38630FF", #ILC
  "#018571",  #monocyte
  "#9E3D22",  #DC
  "#003464FF", #T cells
  "#2C5724FF", #platelets
  "#F8D564FF" #plasma
)
```


```{r}
umap_df %>% ggplot(aes(x=V1, y=V2, color=cell_group)) + geom_point() + theme_bw(base_size = 20) + theme(legend.position = "none") + scale_color_manual(values = my_palette)
#10x10 L
```


```{r}
#cells plot in alphabetical order
my_palette <- c(
"#B89CBF", 
"#F38630FF", 
"#018571", 
"#9E3D22", 
"#003464FF", 
"#2C5724FF", 
"#90659A", 
"#4F3855FF", 
"#01D0B1", 
"#F8D564FF",
"#8AC6FF", 
"#2E9AFF", 
"#0084FF", 
"#006CD1", 
"#5CB0FF", 
"#0054A3",
"#DA7052"
)
```


```{r}
umap_df %>% ggplot(aes(x=V1, y=V2, color=cell_type_low)) + geom_point() + theme_bw(base_size = 20) + scale_color_manual(values = my_palette) + theme(legend.position = "none")
#10x10 L
```



### compare 10X and OA sc data ---

```{r}
prop10X = readRDS('out_parallel_13oct25_10X.rds')
colnames(prop10X)[1] = "sample_name"

setdiff(colnames(proportions), colnames(prop10X))

propOA = proportions %>% dplyr::select(-Megakaryocytes.platelets)

colnames(propOA) <- paste0("OA_", colnames(propOA))
colnames(prop10X) <- paste0("10X_", colnames(prop10X))

info <- readRDS("meta_final.rds")
info %<>% dplyr::select(sample_name, monocytes_perc_combine, lympho_perc_combine)

propOA = merge(propOA, info, by.x="OA_sample_name", by.y= "sample_name")
prop10X = merge(prop10X, info, by.x="10X_sample_name", by.y= "sample_name")

correlation_matrix <- cor(propOA[, -1], prop10X[, -1])  
#colnames(correlation_matrix) <- cell_type_map[colnames(correlation_matrix)]
#rownames(correlation_matrix) <- cell_type_map[rownames(correlation_matrix)]

```


```{r}
corrplot(correlation_matrix, method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", number.cex = 0.5, mar = c(0, 0, 1, 0), is.corr = FALSE)
```



```{r}
pheatmap(correlation_matrix, cluster_rows = TRUE, cluster_cols = TRUE, display_numbers = TRUE, number_color = "black",
  main = "Cross-correlations between OA and 10X proportions")
```


```{r}
corrplot(cor(propOA[, -1]), method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", number.cex = 0.5, mar = c(0, 0, 1, 0), is.corr = FALSE)
#12x12 L
```

```{r}
corrplot(cor(prop10X[, -1]), method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", number.cex = 0.5, mar = c(0, 0, 1, 0), is.corr = FALSE)
#12x12 L
```



```{r}
propOA$OA_predicted_monocytes = propOA$OA_Classical.monocytes + propOA$OA_Non.classical.monocytes

propOA$OA_predicted_lymphoctyes = propOA$OA_Age.associated.B.cells + propOA$OA_CD16..NK.cells + propOA$OA_Memory.B.cells + propOA$OA_Naive.B.cells + propOA$OA_Regulatory.T.cells + propOA$OA_Tcm.Naive.cytotoxic.T.cells + propOA$OA_Tcm.Naive.helper.T.cells + propOA$OA_Tem.Effector.helper.T.cells + propOA$OA_Tem.Temra.cytotoxic.T.cells + propOA$OA_Tem.Trm.cytotoxic.T.cells + propOA$OA_MAIT.cells

prop10X$'10X_predicted_monocytes' = prop10X$'10X_Classical.monocytes' + prop10X$'10X_Non.classical.monocytes'

prop10X$'10X_predicted_lymphoctyes' = prop10X$'10X_Age.associated.B.cells' + prop10X$'10X_CD16..NK.cells' + prop10X$'10X_Memory.B.cells' + prop10X$'10X_Naive.B.cells' + prop10X$'10X_Regulatory.T.cells' + prop10X$'10X_Tcm.Naive.cytotoxic.T.cells' + prop10X$'10X_Tcm.Naive.helper.T.cells' + prop10X$'10X_Tem.Effector.helper.T.cells' + prop10X$'10X_Tem.Temra.cytotoxic.T.cells' + prop10X$'10X_Tem.Trm.cytotoxic.T.cells' + prop10X$'10X_MAIT.cells'
```


```{r}
corrplot(cor(propOA[,c(18:21)]), method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", mar = c(0, 0, 1, 0), is.corr = FALSE, title = "OA single cell as predictor")
```



```{r}
corrplot(cor(prop10X[,c(18:21)]), method = "color", type = "upper", order = "hclust", tl.col = "black", tl.srt = 45, addCoef.col = "black", mar = c(0, 0, 1, 0), is.corr = FALSE, title = "10X single cell as predictor")
```


```{r}
correlation_matrix <- cor(propOA[,c(18:21)], prop10X[,c(18:21)])  
pheatmap(correlation_matrix, cluster_rows = TRUE, cluster_cols = TRUE, display_numbers = TRUE, number_color = "black",
  main = "Cross-correlations between OA and 10X proportions")
```


### twas ---

```{r}
rm(list = ls())
```

get gene/disease pairs from ptwas 
```{r}
#read in gene/disease pairs 
gene_disease = read.delim("~/Desktop/vandy/lea lab/oa/el_cl/figures/13059_2020_2026_MOESM3_ESM.txt", sep=' ', header = F)
gene_disease = gene_disease[-1,-c(5:9)]
colnames(gene_disease) = c("trait", "gene", "most_sig_tissue", "multi_tissue_pval")

#need to add in gene name 
ens_ids <- gene_disease$gene
ens_ids <- sub("\\..*$", "", ens_ids)  # remove anything after the first dot
symbols <- mapIds(
  org.Hs.eg.db,
  keys = ens_ids,
  keytype = "ENSEMBL",
  column = "SYMBOL",
  multiVals = "first")

gene_disease$id_shortened = sub("\\..*$", "", gene_disease$gene)
gene_disease$gene_name <- symbols[gene_disease$id_shortened]

#remove entries that don't have gene names 
gene_disease_sub = gene_disease %>% filter(!is.na(gene_name)) 

# filter 
inclusion = read_excel("~/Desktop/vandy/lea lab/oa/el_cl/figures/post_wgs_resolution/twas_inclusion.xlsx")
inclusion %<>% filter(include==1)
gene_disease_sub %<>% filter(trait %in% inclusion$ID)
```


get list of our genes 
```{r}
model_out <- readRDS("lmm_grm_evryne_grp_fixed_effect_13nov25.rds")

model_out$gene = rownames(model_out)

model_out$padj_urb = p.adjust(model_out$p_value_urb_scale, method = "fdr")
model_out$padj_early = p.adjust(model_out$p_value_early_scale, method = "fdr")

urb_genes = model_out %>% filter(padj_urb<0.1)
early_genes = model_out %>% filter(padj_early<0.1)

current_urb = urb_genes %>% filter(beta_urb_scale>0) %>% dplyr::select(beta_urb_scale, gene)
current_rural = urb_genes %>% filter(beta_urb_scale<0) %>% dplyr::select(beta_urb_scale, gene)
early_urb = early_genes %>% filter(beta_early_scale>0) %>% dplyr::select(beta_early_scale, gene)
early_rural = early_genes %>% filter(beta_early_scale<0) %>% dplyr::select(beta_early_scale, gene)
```

get background gene set 
```{r}
gtex = read.delim('~/Downloads/gtex_v8_cis_qtls_mashr/gtex_v8.eqtls.all_top.z_lfsr.txt')
gtex_genes = unique(gtex$gene)
gtex_genes = data.frame(unique(gtex$gene))
ens_ids <- gtex_genes$unique.gtex.gene.
ens_ids <- sub("\\..*$", "", ens_ids)  
symbols <- mapIds(org.Hs.eg.db, keys = ens_ids, keytype = "ENSEMBL", column = "SYMBOL", multiVals = "first")

gtex_genes$id_shortened = sub("\\..*$", "", gtex_genes$unique.gtex.gene.)
gtex_genes$gene_name <- symbols[gtex_genes$id_shortened]

gtex_genes_sub = gtex_genes %>% filter(!is.na(gene_name))
all_genes = gtex_genes_sub %>% filter(gene_name %in% model_out$gene)
```

add in our genes 
```{r}
all_genes$current_rural <- ifelse(all_genes$gene_name %in% current_rural$gene, 1, 0)
all_genes$current_urb <- ifelse(all_genes$gene_name %in% current_urb$gene, 1, 0)
all_genes$early_rural <- ifelse(all_genes$gene_name %in% early_rural$gene, 1, 0)
all_genes$early_urb <- ifelse(all_genes$gene_name %in% early_urb$gene, 1, 0)
```

```{r}
gene_disease_sub$indicator <- 1

trait_wide <- gene_disease_sub %>%
  distinct(trait, gene_name, indicator) %>%
  tidyr::pivot_wider(
    names_from = trait,
    values_from = indicator,
    values_fill = 0)

all_genes <- all_genes %>%
  left_join(trait_wide, by = c("gene_name"))

all_genes[is.na(all_genes)] <- 0
```


run enrichment 
```{r}
type_cols   <- c("current_rural", "current_urb", "early_rural", "early_urb")
trait_cols  <- colnames(all_genes)[8:109]

get_counts <- function(type_vec, trait_vec) {
  overlap        <- sum(type_vec == 1 & trait_vec == 1)
  only_in_type   <- sum(type_vec == 1 & trait_vec == 0)
  only_in_trait  <- sum(type_vec == 0 & trait_vec == 1)
  neither        <- sum(type_vec == 0 & trait_vec == 0)
  
  list(overlap = overlap,
       only_in_type = only_in_type,
       only_in_trait = only_in_trait,
       neither = neither)
}

results <- list()

for (t in type_cols) {
  for (tr in trait_cols) {
    
    counts <- get_counts(all_genes[[t]], all_genes[[tr]])
    
    ft <- fisher.test(matrix(c(counts$overlap, counts$only_in_type, counts$only_in_trait, counts$neither), nrow = 2,byrow = TRUE)) 
    
    results[[paste(t, tr, sep = "_vs_")]] <- data.frame(
      type_column   = t,
      trait_column  = tr,
      overlap       = counts$overlap,
      only_in_type  = counts$only_in_type,
      only_in_trait = counts$only_in_trait,
      neither       = counts$neither,
      p_value       = ft$p.value,
      odds_ratio    = unname(ft$estimate) ) } }

results_df <- do.call(rbind, results)
results_df$p_adj = p.adjust(results_df$p_value, method = "fdr")
```






plot
```{r}
top5_df <- results_df %>%
  filter(log2(odds_ratio) > 0) %>%
  group_by(type_column) %>%
  arrange(p_adj, .by_group = TRUE) %>%
  slice_head(n = 5) %>%
  ungroup()

top5_df <- top5_df %>%
  mutate(
    trait_type = paste(trait_column, type_column, sep = "_"),
    log2_OR = log2(odds_ratio))

top5_df <- top5_df %>%
  arrange(type_column, log2_OR) %>%
  mutate(trait_type = factor(trait_type, levels = trait_type))

ggplot(top5_df, aes(x = trait_type, y = log2_OR, fill = type_column)) + geom_col(color = "black") +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 20),
        axis.text.y = element_text(size = 30),
        legend.text = element_text(size = 30),
        text = element_text(size = 30)) +
  scale_x_discrete(labels = top5_df$trait_column) +
  labs(x = "Trait", y = "log2(Odds Ratio)") +
  scale_fill_manual(values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) 

```








