---
title: "el_cl_plotting"
author: "layla"
date: "2025-10-08"
output: html_document
---

```{r}
setwd("~/Desktop/vandy/lea lab/oa/el_cl/figures/")
```


packages 
```{r message=FALSE, warning=FALSE}
library(readr)
library(magrittr)
library(dplyr)
#library(variancePartition) for R 4.5 only 
library(ggplot2)
library(igraph)
library(ggraph)
library(clusterProfiler)
library(org.Hs.eg.db)
library(enrichplot)
organism = "org.Hs.eg.db"
library(organism, character.only = TRUE)
library(ggrepel)
library(purrr)
library(broom)
library(tibble)
library(stringr)
library(stringdist)
library(RColorBrewer)
library(scales)
library(colorspace)
library(ggmap)
```


age and sex plot 
-------------------------------------
```{r}
meta <- readRDS("meta_final.rds")
```

```{r}
meta %>% ggplot(aes(x=age, fill=rnaseq_sex)) + geom_density(alpha=0.8) + theme_bw(base_size = 20) + scale_fill_manual(values = c("black","goldenrod2")) + labs(x = "Age", y="Density", fill="Sex")

#L 8x5
```


map and EL vs CL plot
--------------------------------------

```{r}
meta <- readRDS("meta_final.rds")
```

```{r}
meta %>% ggplot(aes(x=Early_urban_PC1, y=urb_score.y)) + geom_point(color="purple4") + theme_classic(base_size = 20) + labs(x = "Early Life Urbanicity", y = "Current Life Urbanicity")
#L 6x5
```

```{r}
OA_merged=read.delim('~/Downloads/18apr25_OA_merged_filt.txt')
medianlat <- median(OA_merged$lat, na.rm = TRUE)
medianlong <- median(OA_merged$long, na.rm = TRUE)

pal=c("#5495CFFF", "#F5AF4DFF", "#DB4743FF", "#7C873EFF", "#FEF4D5FF")

register_google('AIzaSyBqRWaHZTtsvaSOIwDdHjmTVPD-cXvaKrE')
s <- "element:geometry%7Ccolor:0xf5f5f5&style=element:labels%7Cvisibility:off&style=element:labels.icon%7Cvisibility:off&style=element:labels.text.fill%7Ccolor:0x616161&style=element:labels.text.stroke%7Ccolor:0xf5f5f5&style=feature:administrative%7Celement:geometry%7Cvisibility:off&style=feature:administrative.country%7Celement:geometry.stroke%7Ccolor:0x000000%7Cvisibility:on&style=feature:administrative.land_parcel%7Cvisibility:off&style=feature:administrative.land_parcel%7Celement:labels.text.fill%7Ccolor:0xbdbdbd&style=feature:administrative.neighborhood%7Cvisibility:off&style=feature:poi%7Cvisibility:off&style=feature:poi%7Celement:geometry%7Ccolor:0xeeeeee&style=feature:poi%7Celement:labels.text.fill%7Ccolor:0x757575&style=feature:poi.park%7Celement:geometry%7Ccolor:0xe5e5e5&style=feature:poi.park%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&style=feature:road%7Cvisibility:off&style=feature:road%7Celement:geometry%7Ccolor:0xffffff&style=feature:road%7Celement:labels.icon%7Cvisibility:off&style=feature:road.arterial%7Celement:labels.text.fill%7Ccolor:0x757575&style=feature:road.highway%7Celement:geometry%7Ccolor:0xdadada&style=feature:road.highway%7Celement:labels.text.fill%7Ccolor:0x616161&style=feature:road.local%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&style=feature:transit%7Cvisibility:off&style=feature:transit.line%7Celement:geometry%7Ccolor:0xe5e5e5&style=feature:transit.station%7Celement:geometry%7Ccolor:0xeeeeee&style=feature:water%7Celement:geometry%7Ccolor:0xc9c9c9&style=feature:water%7Celement:labels.text.fill%7Ccolor:0x9e9e9e&size=480x360"
c(medianlong, medianlat)

map <- get_googlemap(center=c(102,4), zoom = 7, scale = 1, style = s)
m <- ggmap(map)

Current_urb_by_village<- OA_merged %>%
  group_by(village_id) %>%
  summarize(
    mean_lat= mean(lat),
    mean_long=mean(long),
    mean_urb = mean(urb_score, na.rm=T), 
    count = n() )

Current_urb_by_village<-subset(Current_urb_by_village,count>9)
current_urb_filt = Current_urb_by_village[-which(Current_urb_by_village$village_id==11),]
current_urb_plot<- m +
  geom_point(data = current_urb_filt, aes(x = mean_long, y = mean_lat, fill=mean_urb, size=count), color="black", alpha = 0.8, shape=21) +
  labs(x="Longitude", y="Latitude", size="Sample \n  size", fill="    Adult \n urbanicity \n    score") +
  scale_fill_gradient(low = "#5CB9F4", high = pal[3], breaks = c(2.984, 41.729), labels = c("Low", "High")) + 
  scale_size_continuous(range = c(5,7)) + 
  xlim(100.5, 103.5) +
  ylim(2.5, 6) +
  theme_bw(13); current_urb_plot

#L 5x5
```



tf code
--------------------------------------

```{r}
rm(list = ls())
```


load TF-target interaction list (from TRRUST)
```{r}
tf_targets <- read.delim("trrust_rawdata.human.tsv", header=FALSE, stringsAsFactors=FALSE)
colnames(tf_targets) <- c("TF", "Target", "Mode", "PubMed")
tf_targets %<>% filter(Mode %in% c("Activation", "Repression"))
```


load gene data 
```{r}
model_out = readRDS("model_out_all_24june25.rds")

model_out$every_urb = model_out$`every_beta_as.numeric(scale(meta$urb_score.y))`
model_out$every_early = model_out$`every_beta_as.numeric(scale(meta$Early_urban_PC1))`
model_out$every_age = model_out$every_beta_age_scale
model_out$every_sex = model_out$every_beta_sex_medicalmale

model_out$padj_urb = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$urb_score.y))`, method = "fdr")
model_out$padj_early = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`, method = "fdr")


subset = model_out %>% filter(padj_urb<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0))))
urb_genes = subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0)))

subset = model_out %>% filter(padj_early<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0))))
early_genes = subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0)))

current_urb = urb_genes %>% filter(every_urb>0) %>% dplyr::select(every_urb, gene)
current_rural = urb_genes %>% filter(every_urb<0) %>% dplyr::select(every_urb, gene)
early_urb = early_genes %>% filter(every_early>0) %>% dplyr::select(every_early, gene)
early_rural = early_genes %>% filter(every_early<0) %>% dplyr::select(every_early, gene)
```


create a list of target genes for each TF
```{r}
tf_list <- split(tf_targets$Target, tf_targets$TF)

#filter to be genes that are significant in any model 
gene_list = c(urb_genes$gene, early_genes$gene)
gene_list=gene_list[!duplicated(gene_list)]

filtered_tf_list <- lapply(tf_list, function(genes) {
  intersect(genes, gene_list)})

filtered_tf_list <- Filter(length, filtered_tf_list)

# Filter TFs to those that:
# 1. Have >2 targets remaining
# 2. Are themselves also in the gene_list
filtered_tf_list <- filtered_tf_list[names(filtered_tf_list) %in% model_out$gene]
filtered_tf_list <- Filter(function(targets) length(targets) > 2, filtered_tf_list)

# get average number of targets per TF
summary(sapply(filtered_tf_list, length))
hist(sapply(filtered_tf_list, length), main="Targets per TF", xlab="Number of Targets", 20)

```


adding a filter so that it groups overlapping TFs
```{r}
tf_targets <- read.delim("trrust_rawdata.human.tsv", header=FALSE, stringsAsFactors=FALSE)
colnames(tf_targets) <- c("TF", "Target", "Mode", "PubMed")


model_out = readRDS("model_out_all_24june25.rds")
model_out$every_urb = model_out$`every_beta_as.numeric(scale(meta$urb_score.y))`
model_out$every_early = model_out$`every_beta_as.numeric(scale(meta$Early_urban_PC1))`
model_out$every_age = model_out$every_beta_age_scale
model_out$every_sex = model_out$every_beta_sex_medicalmale

model_out$padj_urb = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$urb_score.y))`, method = "fdr")
model_out$padj_early = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`, method = "fdr")


subset = model_out %>% filter(padj_urb<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0))))
urb_genes = subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0)))

subset = model_out %>% filter(padj_early<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0))))
early_genes = subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0)))

current_urb = urb_genes %>% filter(every_urb>0) %>% dplyr::select(every_urb, gene)
current_rural = urb_genes %>% filter(every_urb<0) %>% dplyr::select(every_urb, gene)
early_urb = early_genes %>% filter(every_early>0) %>% dplyr::select(every_early, gene)
early_rural = early_genes %>% filter(every_early<0) %>% dplyr::select(every_early, gene)


tf_list <- split(tf_targets$Target, tf_targets$TF)



#filter to be genes that are significant in any model 
gene_list = c(urb_genes$gene, early_genes$gene)
gene_list=gene_list[!duplicated(gene_list)]

filtered_tf_list <- lapply(tf_list, function(genes) {
  intersect(genes, gene_list)})

filtered_tf_list <- Filter(length, filtered_tf_list)


# Filter TFs to those that:
# 1. Have >2 targets remaining
# 2. Are themselves also in the gene_list
filtered_tf_list <- filtered_tf_list[names(filtered_tf_list) %in% model_out$gene]
filtered_tf_list <- Filter(function(targets) length(targets) > 3, filtered_tf_list)

# get average number of targets per TF
summary(sapply(filtered_tf_list, length))
hist(sapply(filtered_tf_list, length), main="Targets per TF", xlab="Number of Targets", 20)


```


similarity function 
```{r}
# --- 1. Jaccard similarity function ---
jaccard_sim <- function(a, b) {
  length(intersect(a, b)) / length(union(a, b))
}

tf_names <- names(filtered_tf_list)

# --- 2. Compute similarity matrix ---
sim_mat <- matrix(0, nrow = length(tf_names), ncol = length(tf_names),
                  dimnames = list(tf_names, tf_names))

for (i in seq_along(tf_names)) {
  for (j in seq_along(tf_names)) {
    if (i < j) {
      sim_val <- jaccard_sim(filtered_tf_list[[i]], filtered_tf_list[[j]])
      sim_mat[i, j] <- sim_val
      sim_mat[j, i] <- sim_val
    }
  }
}

# --- 3. Convert similarity to distance ---
dist_mat <- as.dist(1 - sim_mat)  # 1 - Jaccard similarity

# --- 4. Hierarchical clustering ---
hc <- hclust(dist_mat, method = "average")  # UPGMA

# --- 5. Cut tree at Jaccard ≥ 0.8 (distance ≤ 0.2) ---
groups <- cutree(hc, h = 0.7)

# --- 6. Collapse TFs within each cluster ---
collapsed_list <- lapply(split(tf_names, groups), function(tf_group) {
  unique(unlist(filtered_tf_list[tf_group]))
})

# Optional: give merged names that reflect all TFs in the cluster
names(collapsed_list) <- sapply(split(tf_names, groups), paste, collapse = "_")

# --- 7. Check new distribution ---
summary(sapply(collapsed_list, length))
hist(sapply(collapsed_list, length), 
     main = "Targets per (Collapsed) TF", 
     xlab = "Number of Targets", 
     breaks = 20)
```



```{r}
total_background <- 9993  # Total number of genes in background

# Urban
your_genes <- current_urb$gene
results <- lapply(names(collapsed_list), function(tf) {
  tf_genes <- unique(collapsed_list[[tf]])
  overlap <- length(intersect(your_genes, tf_genes))
  only_in_list <- length(setdiff(your_genes, tf_genes))
  only_in_tf <- length(setdiff(tf_genes, your_genes))
  neither <- total_background - (overlap + only_in_list + only_in_tf)
  expected_overlap <- (length(your_genes) * length(tf_genes)) / total_background
  fold_enrichment <- if (expected_overlap > 0) overlap / expected_overlap else NA
  fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
  fisher_test <- fisher.test(fisher_matrix)
  odds_ratio <- unname(fisher_test$estimate)
  data.frame(
    TF = tf,
    p.value = fisher_test$p.value,
    odds_ratio = odds_ratio,
    overlap = overlap,
    total_targets = length(tf_genes),
    expected = expected_overlap,
    fold_enrichment = fold_enrichment
  )
})
current_urb_tf <- do.call(rbind, results)
current_urb_tf$adj.p.value <- p.adjust(current_urb_tf$p.value, method = "BH")

# Rural
your_genes <- current_rural$gene
results <- lapply(names(collapsed_list), function(tf) {
  tf_genes <- unique(collapsed_list[[tf]])
  overlap <- length(intersect(your_genes, tf_genes))
  only_in_list <- length(setdiff(your_genes, tf_genes))
  only_in_tf <- length(setdiff(tf_genes, your_genes))
  neither <- total_background - (overlap + only_in_list + only_in_tf)
  expected_overlap <- (length(your_genes) * length(tf_genes)) / total_background
  fold_enrichment <- if (expected_overlap > 0) overlap / expected_overlap else NA
  fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
  fisher_test <- fisher.test(fisher_matrix)
  odds_ratio <- unname(fisher_test$estimate)
  data.frame(
    TF = tf,
    p.value = fisher_test$p.value,
    odds_ratio = odds_ratio,
    overlap = overlap,
    total_targets = length(tf_genes),
    expected = expected_overlap,
    fold_enrichment = fold_enrichment
  )
})
current_rural_tf <- do.call(rbind, results)
current_rural_tf$adj.p.value <- p.adjust(current_rural_tf$p.value, method = "BH")

# Early Urban
your_genes <- early_urb$gene
results <- lapply(names(collapsed_list), function(tf) {
  tf_genes <- unique(collapsed_list[[tf]])
  overlap <- length(intersect(your_genes, tf_genes))
  only_in_list <- length(setdiff(your_genes, tf_genes))
  only_in_tf <- length(setdiff(tf_genes, your_genes))
  neither <- total_background - (overlap + only_in_list + only_in_tf)
  expected_overlap <- (length(your_genes) * length(tf_genes)) / total_background
  fold_enrichment <- if (expected_overlap > 0) overlap / expected_overlap else NA
  fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
  fisher_test <- fisher.test(fisher_matrix)
  odds_ratio <- unname(fisher_test$estimate)
  data.frame(
    TF = tf,
    p.value = fisher_test$p.value,
    odds_ratio = odds_ratio,
    overlap = overlap,
    total_targets = length(tf_genes),
    expected = expected_overlap,
    fold_enrichment = fold_enrichment
  )
})
early_urb_tf <- do.call(rbind, results)
early_urb_tf$adj.p.value <- p.adjust(early_urb_tf$p.value, method = "BH")

# Early Rural
your_genes <- early_rural$gene
results <- lapply(names(collapsed_list), function(tf) {
  tf_genes <- unique(collapsed_list[[tf]])
  overlap <- length(intersect(your_genes, tf_genes))
  only_in_list <- length(setdiff(your_genes, tf_genes))
  only_in_tf <- length(setdiff(tf_genes, your_genes))
  neither <- total_background - (overlap + only_in_list + only_in_tf)
  expected_overlap <- (length(your_genes) * length(tf_genes)) / total_background
  fold_enrichment <- if (expected_overlap > 0) overlap / expected_overlap else NA
  fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
  fisher_test <- fisher.test(fisher_matrix)
  odds_ratio <- unname(fisher_test$estimate)
  data.frame(
    TF = tf,
    p.value = fisher_test$p.value,
    odds_ratio = odds_ratio,
    overlap = overlap,
    total_targets = length(tf_genes),
    expected = expected_overlap,
    fold_enrichment = fold_enrichment
  )
})
early_rural_tf <- do.call(rbind, results)
early_rural_tf$adj.p.value <- p.adjust(early_rural_tf$p.value, method = "BH")
```


```{r}
current_urb_tf$type = "current_urb"
current_rural_tf$type = "current_rural"
early_urb_tf$type = "early_urb"
early_rural_tf$type = "early_rural"

combined_df = rbind(current_urb_tf, current_rural_tf, early_urb_tf, early_rural_tf)
```

```{r}
combined_df_filt = combined_df %>% filter(!odds_ratio=="Inf")
combined_df_filt = combined_df_filt %>% filter(p.value < 0.1)
combined_df_filt = combined_df_filt %>% mutate(significance = ifelse(adj.p.value < 0.1, "p.adj < 0.1", "p.value < 0.1"))

combined_df_filt <- combined_df_filt %>%
  mutate(TF_type = paste(TF, type, sep = "_")) %>%
  arrange(type, desc(fold_enrichment)) %>%
  mutate(TF_type = factor(TF_type, levels = unique(TF_type)))

combined_df_filt$TF_label <- combined_df_filt$TF 
```



making sure groups are unique
```{r}
# Extract vectors of genes
g1 <- current_rural$gene
g2 <- current_urb$gene
g3 <- early_rural$gene
g4 <- early_urb$gene

# Combine into a list
gene_lists <- list(current_rural = g1, current_urb = g2, early_rural = g3, early_urb = g4)

# Which genes appear in more than one group?
overlapping_genes <- names(table(unlist(gene_lists)))[table(unlist(gene_lists)) > 1]

combn(names(gene_lists), 2, function(x) {
  overlap <- intersect(gene_lists[[x[1]]], gene_lists[[x[2]]])
  cat("\nOverlap between", x[1], "and", x[2], ":\n")
  print(overlap)
})
```



re-running but removing overlapping genes 
```{r}
all_gene_lists <- list(
  urb = current_urb$gene,
  rural = current_rural$gene,
  early_urb = early_urb$gene,
  early_rural = early_rural$gene
)

# Find genes that are in >1 group
gene_counts <- table(unlist(all_gene_lists))
overlapping_genes <- names(gene_counts[gene_counts > 1])
```



```{r}
run_tf_enrichment <- function(gene_vector, tf_list, total_background, exclude_genes = NULL) {
  # Remove overlapping genes if provided
  if (!is.null(exclude_genes)) {
    gene_vector <- setdiff(gene_vector, exclude_genes)
  }
  
  results <- lapply(names(tf_list), function(tf) {
    tf_genes <- unique(tf_list[[tf]])
    
    overlap <- length(intersect(gene_vector, tf_genes))
    only_in_list <- length(setdiff(gene_vector, tf_genes))
    only_in_tf <- length(setdiff(tf_genes, gene_vector))
    neither <- total_background - (overlap + only_in_list + only_in_tf)
    
    expected_overlap <- (length(gene_vector) * length(tf_genes)) / total_background
    fold_enrichment <- if (expected_overlap > 0) overlap / expected_overlap else NA
    
    fisher_matrix <- matrix(c(overlap, only_in_list, only_in_tf, neither), nrow = 2)
    fisher_test <- fisher.test(fisher_matrix)
    odds_ratio <- unname(fisher_test$estimate)
    
    data.frame(
      TF = tf,
      p.value = fisher_test$p.value,
      odds_ratio = odds_ratio,
      overlap = overlap,
      total_targets = length(tf_genes),
      expected = expected_overlap,
      fold_enrichment = fold_enrichment
    )
  })
  
  results_df <- do.call(rbind, results)
  results_df$adj.p.value <- p.adjust(results_df$p.value, method = "BH")
  return(results_df)
}
```



```{r}
total_background <- 9993  # total number of genes in background

current_urb_tf   <- run_tf_enrichment(current_urb$gene,   collapsed_list, total_background, exclude_genes = overlapping_genes)
current_rural_tf <- run_tf_enrichment(current_rural$gene, collapsed_list, total_background, exclude_genes = overlapping_genes)
early_urb_tf     <- run_tf_enrichment(early_urb$gene,     collapsed_list, total_background, exclude_genes = overlapping_genes)
early_rural_tf   <- run_tf_enrichment(early_rural$gene,   collapsed_list, total_background, exclude_genes = overlapping_genes)
```


```{r}
current_urb_tf$type = "current_urb"
current_rural_tf$type = "current_rural"
early_urb_tf$type = "early_urb"
early_rural_tf$type = "early_rural"

combined_df = rbind(current_urb_tf, current_rural_tf, early_urb_tf, early_rural_tf)
```

```{r}
combined_df_filt = combined_df %>% filter(!odds_ratio=="Inf")
combined_df_filt = combined_df_filt %>% filter(p.value < 0.1)
combined_df_filt = combined_df_filt %>% mutate(significance = ifelse(adj.p.value < 0.1, "p.adj < 0.1", "p.value < 0.1"))

combined_df_filt <- combined_df_filt %>%
  mutate(TF_type = paste(TF, type, sep = "_")) %>%
  arrange(type, desc(fold_enrichment)) %>%
  mutate(TF_type = factor(TF_type, levels = unique(TF_type)))

combined_df_filt$TF_label <- combined_df_filt$TF 
```



```{r}
top5_df <- combined_df_filt %>%
  group_by(type) %>%
  arrange(adj.p.value, .by_group = TRUE) %>%
  slice_head(n = 7) %>% ungroup()

ggplot(top5_df, aes(x = TF_type, y = log2(odds_ratio), fill = type)) +
  geom_col(color = "black") +
  scale_x_discrete(labels = top5_df$TF_label) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 20),
    axis.text.y = element_text(size = 20), legend.text = element_text(size = 20), text = element_text(size = 20)) +
  scale_fill_manual(values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  labs(x = "TF", y = "log2(Odds Ratio)")
```


chart showing TF sharing 
```{r}
top5_df <- combined_df_filt %>%
  group_by(type) %>%
  arrange(adj.p.value, .by_group = TRUE) %>%
  slice_head(n = 5) %>%
  ungroup()

top5_df <- top5_df %>%
  mutate(TF = factor(TF, levels = unique(TF)))

ggplot(top5_df, aes(x = type, y = TF, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(
    values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  theme_minimal() +
  labs(title = "Top 5 TFs per Category", x = "Type", y = "Transcription Factor") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), panel.grid = element_blank())

#L 5x5
```
```{r}
ggplot(top5_df, aes(x = type, y = TF, fill = type, alpha = fold_enrichment)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(
    values = c("darkolivegreen", "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  scale_alpha_continuous(range = c(0.5, 1), name = "Fold Enrichment") +
  theme_minimal() +
  labs(title = "Top 5 TFs per Category", x = "Type", y = "Transcription Factor") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), panel.grid = element_blank())
```



looking at nfkb1_rela targets 
```{r}
# Pull NFKB1_RELA target genes from your collapsed_list
nfkb_targets <- unique(collapsed_list[["NFKB1_RELA"]])

# For each category, find which genes overlap
nfkb_df <- bind_rows(
  data.frame(gene = intersect(current_urb$gene, nfkb_targets),   type = "current_urb"),
  data.frame(gene = intersect(current_rural$gene, nfkb_targets), type = "current_rural"),
  data.frame(gene = intersect(early_urb$gene, nfkb_targets),     type = "early_urb"),
  data.frame(gene = intersect(early_rural$gene, nfkb_targets),   type = "early_rural"))
```


dotplot/list style 
```{r}
ggplot(nfkb_df, aes(x = type, y = gene, color = type)) +
  geom_point(size = 4, alpha = 0.8) +
  scale_color_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2"
  )) +
  theme_minimal() +
  labs(
    title = "Genes regulated by NFKB1_RELA",
    x = "Category",
    y = "Gene"
  ) +
  theme(axis.text.y = element_text(size = 6))

#p 5x5 
```


adding in directionality 
```{r}
nfkb_rela_targets <- tf_targets %>%
  filter(TF %in% c("NFKB1", "RELA"))

nfkb_annot <- nfkb_df %>%
    inner_join(nfkb_rela_targets, by = c("gene" = "Target")) %>%
    group_by(gene, type) %>%
    summarise(
        TF = paste(unique(TF), collapse = "+"),
        direction = paste(unique(Mode), collapse = "+"),
        .groups = "drop")

ggplot(nfkb_annot, aes(x = type, y = gene, color = type, shape = direction)) +
  geom_point(size = 4) +
  scale_color_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2")) + 
  scale_fill_manual(values = c( "darkolivegreen",  "darkolivegreen3", "dodgerblue3", "skyblue2")) +
  scale_shape_manual(values = c(
    "Activation"             = 24,  # triangle up
    "Repression"             = 25,  # triangle down
    "Unknown"                = 21,  # circle
    "Activation+Repression"  = 23,  # diamond
    "Activation+Unknown"     = 22,  # square
    "Repression+Unknown"     = 8    # star
  )) +
  theme_minimal() +
  labs(
    title = "NFKB1_RELA target genes by category",
    x = "Category",
    y = "Gene",
    shape = "Direction"
  ) +
  theme(axis.text.y = element_text(size = 6))


ggplot(nfkb_annot, aes(x = type, y = gene)) +
  geom_point(aes(shape = direction, fill = type), size = 4, color = "black", alpha = 0.9) +
  scale_shape_manual(values = c(
    "Activation"             = 24,  # triangle up
    "Repression"             = 25,  # triangle down
    "Unknown"                = 21,  # circle
    "Activation+Repression"  = 23,  # diamond
    "Activation+Unknown"     = 22,  # square
    "Repression+Unknown"     = 8    # star
  )) +
  scale_fill_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2"
  )) +
  theme_minimal() +
  labs(
    title = "NFKB1_RELA target genes by category",
    x = "Category",
    y = "Gene",
    shape = "Direction",
    fill = "Category"
  ) +
  theme(axis.text.y = element_text(size = 6))
```

block style 
```{r}
ggplot(nfkb_df, aes(x = type, y = gene, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2")) +
  theme_minimal() +
  labs(title = "Genes regulated by NFKB1_RELA", x = "Category", y = "Gene") +
  theme(axis.text.y = element_text(size = 6))

#p 5x5 
```

addinging heirarchial clustering 
```{r}
nfkb_df <- bind_rows(
  current_urb  %>% filter(gene %in% nfkb_targets)  %>% transmute(gene, type = "current_urb",  beta = every_urb),
  current_rural %>% filter(gene %in% nfkb_targets) %>% transmute(gene, type = "current_rural", beta = every_urb),
  early_urb   %>% filter(gene %in% nfkb_targets)   %>% transmute(gene, type = "early_urb",   beta = every_early),
  early_rural %>% filter(gene %in% nfkb_targets)   %>% transmute(gene, type = "early_rural", beta = every_early))
```

```{r}
mat <- nfkb_df %>%
  tidyr::pivot_wider(names_from = type, values_from = beta, values_fill = 0) %>%
  column_to_rownames("gene")

# cluster rows (genes) and columns (categories)
row_order <- hclust(dist(mat))$order
col_order <- hclust(dist(t(mat)))$order

# apply ordering
nfkb_df$gene <- factor(nfkb_df$gene, levels = rownames(mat)[row_order])
nfkb_df$type <- factor(nfkb_df$type, levels = colnames(mat)[col_order])

# plot
ggplot(nfkb_df, aes(x = type, y = gene, fill = type, alpha = beta)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2")) + 
  scale_alpha_continuous(range = c(0.6, 1), name = "Effect Size") +
  theme_minimal() +
  labs(title = "NFKB1_RELA Targets (clustered)", x = "Category", y = "Gene") +
  theme(axis.text.y = element_text(size = 6))


ggplot(nfkb_df, aes(x = type, y = gene, fill = type)) +
  geom_tile(color = "white", width = 0.9, height = 0.9) +
  scale_fill_manual(values = c(
    "current_rural" = "darkolivegreen",
    "current_urb"   = "darkolivegreen3",
    "early_rural"   = "dodgerblue3",
    "early_urb"     = "skyblue2")) + 
  theme_minimal(base_size = 20) +
  labs(title = "NFKB1_RELA Targets (clustered)", x = "Category", y = "Gene")
```


urb rural enrich plot 
------------------------------------

```{r}
rm(list = ls())
```

```{r}
set.seed(24)
```

read in model results 
```{r}
model_out <- readRDS("model_out_all_24june25.rds")

model_out$every_urb = model_out$`every_beta_as.numeric(scale(meta$urb_score.y))`
model_out$every_early = model_out$`every_beta_as.numeric(scale(meta$Early_urban_PC1))`
model_out$every_age = model_out$every_beta_age_scale
model_out$every_sex = model_out$every_beta_sex_medicalmale

model_out$padj_urb = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$urb_score.y))`, method = "fdr")
model_out$padj_early = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`, method = "fdr")

subset = model_out %>% filter(padj_urb<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0))))
urb_genes = subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0)))

subset = model_out %>% filter(padj_early<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0))))
early_genes = subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0)))
```


CL urban, background (all genes), test (all CL urban genes)
```{r}
test= subset(urb_genes, urb_genes$every_urb>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
dotplot(filtered_urban, showCategory=10, font.size=10)
```


CL rural, background (all genes), test (all CL rural genes)
```{r}
test= subset(urb_genes, urb_genes$every_urb<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
dotplot(filtered_rural, showCategory=10, font.size=10)
```


CL urban, background (all genes), test (significant CL urban genes)
```{r}
test= subset(model_out, model_out$padj_urb<0.1 & model_out$every_urb>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
dotplot(filtered_urban, showCategory=10, font.size=10)
```




CL rural, background (all genes), test (significant CL rural genes)
```{r}
test= subset(model_out, model_out$padj_urb<0.1 & model_out$every_urb<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
dotplot(filtered_rural, showCategory=10, font.size=10)
```



```{r}
# Add a column to each df to mark Urban vs Rural
urban_df <- data.frame(filtered_urban %>% mutate(Group = "Urban"))
rural_df <- data.frame(filtered_rural %>% mutate(Group = "Rural"))

# Take top 5 from each based on adjusted p-value
urban_top5 <- urban_df %>% arrange(p.adjust) %>% slice_head(n = 5)
rural_top5 <- rural_df %>% arrange(p.adjust) %>% slice_head(n = 5)

df_top <- bind_rows(urban_top5, rural_top5) %>%
  mutate(Description_clean = Description)

df_top <- df_top %>%
  group_by(Group) %>%
  arrange(desc(FoldEnrichment), .by_group = TRUE) %>%
  mutate(ordering = row_number()) %>%
  ungroup()


manual_order <- c(
  # Rural top 5
  "regulation of B cell receptor signaling pathway",
  "regulation of antigen receptor-mediated signaling pathway",
  "antigen receptor-mediated signaling pathway",
  "regulation of leukocyte activation",
  "T cell activation",
  
  # Urban top 5
  "small molecule catabolic process",
  "carboxylic acid metabolic process",
  "purine-containing compound metabolic process",
  "nucleoside phosphate metabolic process",
  "nucleobase-containing small molecule metabolic process")

# Apply to factor
df_top <- df_top %>%
  mutate(Description_clean = factor(Description_clean, levels = manual_order))
```

```{r}
ggplot(df_top, aes(x = Description_clean, y = 0)) +
  geom_point(aes(
    size = Count,
    color = Group,
    alpha = FoldEnrichment)) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "#17154FFF", "Urban" = "#D95E31FF")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group",
    alpha = "Fold Enrichment")

#L 6x5
```





EL urban, background (all genes), test (all EL urban genes)
```{r}
test= subset(early_genes, early_genes$every_early>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
dotplot(GO, showCategory=10, font.size=10)
```


EL rural, background (all genes), test (all EL rural genes)
```{r}
test= subset(early_genes, early_genes$every_early<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
dotplot(GO, showCategory=10, font.size=10)
```



EL urban, background (all genes), test (significant EL urban genes)
```{r}
test= subset(model_out, model_out$padj_early<0.1 & model_out$every_early>0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_urban <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_urban)
dotplot(GO, showCategory=10, font.size=10)
```

EL rural, background (all genes), test (significant EL rural genes)
```{r}
test= subset(model_out, model_out$padj_early<0.1 & model_out$every_early<0)$gene
back<-model_out$gene

GO <- enrichGO(gene = test,
                      universe = back,
                      OrgDb = org.Hs.eg.db,
                      ont  = "BP", keyType = "SYMBOL",
                      pAdjustMethod = "none",
                      qvalueCutoff  = 1,minGSSize = 5, maxGSSize = 500,
                      readable = TRUE)

filtered_df <- GO@result[GO@result$p.adjust < 0.1, ]

filtered_rural <- new("enrichResult",
                          result = filtered_df,
                          pvalueCutoff = 1,
                          pAdjustMethod = "BH",
                          qvalueCutoff = 1,
                          organism = "UNKNOWN", 
                          ontology = "BP",
                          gene = GO@gene,
                          universe = GO@universe,
                          geneSets = GO@geneSets,
                          keytype = "UNKNOWN",  
                          readable = TRUE)
edo <- pairwise_termsim(filtered_rural)
dotplot(GO, showCategory=10, font.size=10)
```

```{r}
# Add a column to each df to mark Urban vs Rural
urban_df <- data.frame(filtered_urban %>% mutate(Group = "Urban"))
rural_df <- data.frame(filtered_rural %>% mutate(Group = "Rural"))

# Take top 5 from each based on adjusted p-value
urban_top5 <- urban_df %>% arrange(p.adjust) %>% slice_head(n = 5)
rural_top5 <- rural_df %>% arrange(p.adjust) %>% slice_head(n = 5)

df_top <- bind_rows(urban_top5, rural_top5) %>%
  mutate(Description_clean = Description)

df_top <- df_top %>%
  group_by(Group) %>%
  arrange(desc(FoldEnrichment), .by_group = TRUE) %>%
  mutate(ordering = row_number()) %>%
  ungroup()

manual_order <- c(
  # Rural top 5
  "glycoside catabolic process",
  "norepinephrine secretion",
  "T cell mediated cytotoxicity",
  "regulation of T cell mediated cytotoxicity",
  "bone development",
  
  
  # Urban top 5
  "'de novo' protein folding",
  "chaperone-mediated protein folding",
  "response to unfolded protein",
  "protein folding",
  "cellular response to lipopolysaccharide")

# Apply to factor
df_top <- df_top %>%
  mutate(Description_clean = factor(Description_clean, levels = manual_order))
```

```{r}
ggplot(df_top, aes(x = Description_clean, y = 0)) +
  geom_point(aes(
    size = Count,
    color = Group,
    alpha = FoldEnrichment)) +
  scale_size(range = c(3, 12)) +
  scale_color_manual(
    values = c("Rural" = "#17154FFF", "Urban" = "#D95E31FF")) +
  scale_alpha_continuous(range = c(0.8, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 75, hjust = 1, size = 12),
    panel.grid = element_blank()) +
  labs(
    x = NULL, y = NULL,
    size = "Gene Count",
    color = "Group",
    alpha = "Fold Enrichment")

#L 6x5
```






volcano plots
-------------------------------------------
```{r}
rm(list = ls())
```

```{r}
model_out <- readRDS("model_out_all_24june25.rds")
```

```{r}
model_out$every_urb = model_out$`every_beta_as.numeric(scale(meta$urb_score.y))`
model_out$every_early = model_out$`every_beta_as.numeric(scale(meta$Early_urban_PC1))`
model_out$every_age = model_out$every_beta_age_scale
model_out$every_sex = model_out$every_beta_sex_medicalmale
```

```{r}
model_out$padj_urb = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$urb_score.y))`, method = "fdr")
model_out$padj_early = p.adjust(model_out$`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`, method = "fdr")
```


```{r}
subset = model_out %>% filter(padj_urb<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0))))
urb_genes = subset %>% filter(apply(dplyr::select(., every_urb, `sem_beta_as.numeric(scale(meta$urb_score.y))`, `sen_beta_as.numeric(scale(meta$urb_score.y))`, `pm_beta_as.numeric(scale(meta$urb_score.y))`), 1, function(x) all(x > 0) || all(x < 0)))

subset = model_out %>% filter(padj_early<0.1)
dim(subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0))))
early_genes = subset %>% filter(apply(dplyr::select(., every_early, `sem_beta_as.numeric(scale(meta$Early_urban_PC1))`, `sen_beta_as.numeric(scale(meta$Early_urban_PC1))`, `pm_beta_as.numeric(scale(meta$Early_urban_PC1))`), 1, function(x) all(x > 0) || all(x < 0)))
```



```{r}
model_out$sig_current = ifelse(model_out$padj_urb<0.1, 1, 0)
model_out$sig_early = ifelse(model_out$padj_early<0.1, 1, 0)
```


```{r}
top_genes <- model_out[model_out$sig_current == 1, ]
top_genes <- top_genes[order(top_genes$`every_p_value_as.numeric(scale(meta$urb_score.y))`, -abs(top_genes$every_urb)), ][1:10, ]

model_out %>% ggplot(aes(x=every_urb, y=-log10(`every_p_value_as.numeric(scale(meta$urb_score.y))`))) + geom_point(aes(color = as.factor(sig_current), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "darkgreen"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Current Lifestyle Genes") + ylim(0,10)

#L 5x5
```

```{r}
top_genes_filt = top_genes %>% filter(gene %in% c("PDK4", "LCN10"))

model_out %>% ggplot(aes(x=every_urb, y=-log10(`every_p_value_as.numeric(scale(meta$urb_score.y))`))) + geom_point(aes(color = as.factor(sig_current), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "darkgreen"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes_filt, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Current Lifestyle Genes") + ylim(0,10)

#L 5x5
```


```{r}
top_genes <- model_out[model_out$sig_early == 1, ]
top_genes <- top_genes[order(top_genes$`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`, -abs(top_genes$every_early)), ][1:10, ]

model_out %>% ggplot(aes(x=every_early, y=-log10(`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`))) + geom_point(aes(color = as.factor(sig_early), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "steelblue"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Early Lifestyle Genes") + ylim(0,10)

#L 5x5
```

```{r}
top_genes_filt = top_genes %>% filter(gene %in% c("CXCL1", "CPT2"))

model_out %>% ggplot(aes(x=every_early, y=-log10(`every_p_value_as.numeric(scale(meta$Early_urban_PC1))`))) + geom_point(aes(color = as.factor(sig_early), alpha = 0.25)) + theme_minimal(base_size = 20) + xlab("logfc") + ylab("-log10(p-value)") + scale_color_manual(values = c("grey", "steelblue"))  + theme(legend.position = "none") + geom_label_repel(data = top_genes_filt, aes(label = gene), size = 3, box.padding = 0.8, label.size = 0.2, label.r = unit(0.15, "lines"), max.overlaps = 20) +  ggtitle("Early Lifestyle Genes") + ylim(0,10)

#L 5x5
```


```{r}
gene_list = c(urb_genes$gene, early_genes$gene)
gene_list=gene_list[!duplicated(gene_list)]

plot_data <- model_out %>%
  dplyr::select(every_urb, every_early) %>%
  tidyr::pivot_longer(cols = everything(),names_to = "variable", values_to = "value")

plot_data %>% filter(abs(value) <0.15) %>% ggplot(aes(x = abs(value), fill = variable)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Gene Expression Effect Sizes", x = "abs(Effect Size)", y = "Density")
#L 8x5


plot_data <- model_out %>% filter(gene %in% gene_list) %>%
  dplyr::select(every_urb, every_early) %>%
  tidyr::pivot_longer(cols = everything(),names_to = "variable", values_to = "value")

plot_data %>% filter(abs(value) <0.15) %>% ggplot(aes(x = abs(value), fill = variable)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
#L 5x4
```


cytokines 
-------------------------------------------
```{r}
rm(list = ls())
```

annoyingly the code is dependent on itself for some reason 
trying to figure out but for now just saving model output

```{r}
model_summary_df = readRDS('cytokine_urb_early_30sep25.rds')
```

```{r}
model_summary_df %>%
  filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>%
  ggplot(aes(x = estimate, y = response, 
             fill = direction, alpha = term)) +
  geom_col(position = position_dodge(width = 0.7), width = 0.8) +
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_fill_manual(values = c("pro" = "red4", "anti" = "#1B9E77")) +
  scale_alpha_manual(values = c("scale(urb_score)" = 1, "scale(Early_urban_PC1)" = 0.5)) +
  labs(x = "Beta Estimate", y = "", fill = "Type", alpha = "Term") +
  theme_minimal(base_size = 20)

#L 15x7
```


```{r}

model_summary_df %>% filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>% ggplot(aes(x = abs(estimate), fill = term)) + geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Cytokine Effect Sizes", x = "abs(Effect Size)", y = "Density") 
#L 10x5


model_summary_df %>% filter(term %in% c("scale(urb_score)", "scale(Early_urban_PC1)")) %>% filter(!significance=="ns")  %>% ggplot(aes(x = abs(estimate), fill = term)) + geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
#L 5x4
```





cell type proportions modeling 
----------------------------------------------------------------------
```{r}
rm(list = ls())
```

```{r}
proportions <- read_csv("subset_proportions (1).csv")
colnames(proportions)[1] = "sample_name"

info <- readRDS("meta_final.rds")
info %<>% dplyr::select(urb_score.y, Early_urban_PC1, age_scale, sex_medical, sample_name)

info$urb_score.y = as.numeric(scale(info$urb_score.y))
info$Early_urban_PC1 = as.numeric(scale(info$Early_urban_PC1))
info$sex_medical = ifelse(info$sex_medical=="female", 0, 1)

df = left_join(info, proportions, by="sample_name")
```


```{r}
#list of cell column names 
cell_columns <- c(
  "Age-associated B cells", "B cells", "CD16+ NK cells", "CD16- NK cells", "CD8a/a",
  "CRTAM+ gamma-delta T cells", "Classical monocytes", "Cycling T cells", "DC1", "DC2",
  "Double-negative thymocytes", "Double-positive thymocytes", "Early lymphoid/T lymphoid",
  "Epithelial cells", "Follicular helper T cells", "Germinal center B cells", "HSC/MPP", "ILC3",
  "Intermediate macrophages", "Intestinal macrophages", "MAIT cells", "Macrophages",
  "Megakaryocytes/platelets", "Memory B cells", "Memory CD4+ cytotoxic T cells", "Mid erythroid",
  "Mono-mac", "Monocytes", "NK cells", "NKT cells", "Naive B cells", "Non-classical monocytes",
  "Plasma cells", "Plasmablasts", "Regulatory T cells", "T(agonist)", "Tcm/Naive cytotoxic T cells",
  "Tcm/Naive helper T cells", "Tem/Effector helper T cells", "Tem/Temra cytotoxic T cells",
  "Tem/Trm cytotoxic T cells", "Transitional B cells", "Treg(diff)", "Trm cytotoxic T cells",
  "Type 1 helper T cells", "Type 17 helper T cells", "gamma-delta T cells", "pDC")

#clean up names (gets rid of spaces and symbols)
cell_columns_clean <- cell_columns %>%
  gsub("\\+", "pos", .) %>%
  gsub("-", "neg", .) %>%
  make.names()
```


model and store results 
```{r}
model_results <- setNames(vector("list", length(cell_columns)), cell_columns_clean)

for (i in seq_along(cell_columns)) {
  cell_col <- cell_columns[i]
  model_results[[i]] <- lm(as.formula(paste0("`", cell_col, "` ~ urb_score.y + Early_urban_PC1 + age_scale + sex_medical")), data = df)
}
```



extract coefficients 
```{r}
coeff_df <- bind_rows(
  lapply(names(model_results), function(name) {
    broom::tidy(model_results[[name]]) %>%
      mutate(cell_type = name)
  }),
  .id = "model"
)

coeff_df$p.adj = p.adjust(coeff_df$p.value, method = "fdr")
```


plot results 
```{r}
plot_df <- coeff_df %>%
  filter(term != "(Intercept)") %>%              # remove intercept
  mutate(significance = ifelse(p.adj < 0.1, "p.adj < 0.1", "n.s."))  # add shading category
```



```{r}
cell_groups <- list(
  T_cells = c("CRTAMpos.gammanegdelta.T.cells", "Cycling.T.cells", "Doublenegnegative.thymocytes", 
              "Doublenegpositive.thymocytes", "Early.lymphoid.T.lymphoid", "Follicular.helper.T.cells",
              "gammanegdelta.T.cells", "MAIT.cells", "Memory.CD4pos.cytotoxic.T.cells", "Regulatory.T.cells",
              "T.agonist.", "Tcm.Naive.cytotoxic.T.cells", "Tcm.Naive.helper.T.cells", 
              "Tem.Effector.helper.T.cells", "Tem.Temra.cytotoxic.T.cells", "Tem.Trm.cytotoxic.T.cells",
              "Trm.cytotoxic.T.cells", "Type.1.helper.T.cells", "Type.17.helper.T.cells", "Treg.diff.", "CD8a.a"),
  Monocytes = c("Classical.monocytes", "Intermediate.macrophages", "Intestinal.macrophages", "Macrophages",
                "Monocytes", "Mononegmac", "Nonnegclassical.monocytes", "DC1", "DC2", "pDC"),
  B_cells = c("Agenegassociated.B.cells", "B.cells", "Germinal.center.B.cells", "Memory.B.cells",
              "Naive.B.cells", "Plasma.cells", "Plasmablasts", "Transitional.B.cells"),
  NK_cells = c("CD16pos.NK.cells", "CD16neg.NK.cells", "NK.cells", "NKT.cells"),
  Stem_Progenitor = c("HSC.MPP", "Mid.erythroid", "Megakaryocytes.platelets")
)

plot_df$cell_group <- sapply(plot_df$cell_type, function(ct) {
  group <- names(cell_groups)[sapply(cell_groups, function(x) ct %in% x)]
  if (length(group) == 0) return("Other") else return(group)})

plot_df_filtered <- plot_df %>%
  filter(term %in% c("urb_score.y", "Early_urban_PC1")) %>% filter(!cell_type %in% c("Intestinal.macrophages", "Epithelial.cells", "Trm.cytotoxic.T.cells", "Double.negative.thymocytes", "Double.positive.thymocytes"))

plot_df_ordered <- plot_df_filtered %>%
  group_by(cell_group, cell_type) %>%
  summarize(max_est = max(abs(estimate)), .groups = "drop") %>%
  arrange(factor(cell_group, levels = c("T_cells", "Monocytes", "B_cells", "NK_cells", "Stem_Progenitor", "Other")),
          desc(max_est)) %>% mutate(cell_type = factor(cell_type, levels = unique(cell_type)))

plot_df_filtered$cell_type <- factor(plot_df_filtered$cell_type,
                                     levels = levels(plot_df_ordered$cell_type))

plot_df_filtered$cell_type_clean <- recode(plot_df_filtered$cell_type,
  "Agenegassociated.B.cells" = "Age-Associated B Cells",
  "B.cells" = "B Cells",
  "CD16pos.NK.cells" = "CD16+ NK Cells",
  "CD16neg.NK.cells" = "CD16- NK Cells",
  "CD8a.a" = "CD8 T Cells",
  "Classical.monocytes" = "Classical Monocytes",
  "CRTAMpos.gammanegdelta.T.cells" = "CRTAM+ gamma-delta T Cells",
  "Cycling.T.cells" = "Cycling T Cells",
  "DC1" = "DC1",
  "DC2" = "DC2",
  "Doublenegnegative.thymocytes" = "Double Negative Thymocytes",
  "Doublenegpositive.thymocytes" = "Double Positive Thymocytes",
  "Early.lymphoid.T.lymphoid" = "Early Lymphoid T Cells",
  "Epithelial.cells" = "Epithelial Cells",
  "Follicular.helper.T.cells" = "Follicular Helper T Cells",
  "gammanegdelta.T.cells" = "gamma-delta T Cells",
  "Germinal.center.B.cells" = "Germinal Center B Cells",
  "HSC.MPP" = "HSC/MPP",
  "ILC3" = "ILC3",
  "Intermediate.macrophages" = "Intermediate Macrophages",
  "Intestinal.macrophages" = "Intestinal Macrophages",
  "Macrophages" = "Macrophages",
  "MAIT.cells" = "MAIT Cells",
  "Megakaryocytes.platelets" = "Megakaryocytes/Platelets",
  "Memory.B.cells" = "Memory B Cells",
  "Memory.CD4pos.cytotoxic.T.cells" = "Memory CD4 Cytotoxic T Cells",
  "Mid.erythroid" = "Mid Erythroid",
  "Mononegmac" = "Mono-Mac",
  "Monocytes" = "Monocytes",
  "Naive.B.cells" = "Naive B Cells",
  "NK.cells" = "NK Cells",
  "NKT.cells" = "NKT Cells",
  "Nonnegclassical.monocytes" = "Non-Classical Monocytes",
  "pDC" = "pDC",
  "Plasma.cells" = "Plasma Cells",
  "Plasmablasts" = "Plasmablasts",
  "Regulatory.T.cells" = "Regulatory T Cells",
  "T.agonist." = "T Agonist",
  "Tcm.Naive.cytotoxic.T.cells" = "Tcm/Naive Cytotoxic T Cells",
  "Tcm.Naive.helper.T.cells" = "Tcm/Naive Helper T Cells",
  "Tem.Effector.helper.T.cells" = "Tem/Effector Helper T Cells",
  "Tem.Temra.cytotoxic.T.cells" = "Tem/Temra Cytotoxic T Cells",
  "Tem.Trm.cytotoxic.T.cells" = "Tem/Trm Cytotoxic T Cells",
  "Transitional.B.cells" = "Transitional B Cells",
  "Treg.diff." = "Differentiated Tregs",
  "Trm.cytotoxic.T.cells" = "Trm Cytotoxic T Cells",
  "Type.1.helper.T.cells" = "Type 1 Helper T Cells",
  "Type.17.helper.T.cells" = "Type 17 Helper T Cells"
)
```



```{r}
ggplot(plot_df_filtered, aes(x = cell_type_clean, y = estimate, fill = term, alpha = significance)) +
  geom_col(position = position_dodge(width = 0.8), width = 0.7) +
  scale_alpha_manual(values = c("p.adj < 0.1" = 1, "n.s." = 0.4)) +
  scale_fill_manual(values = c("steelblue", "darkgreen")) +
  labs(x = "Cell Type", y = "Beta Estimate", fill = "Model Term", alpha = "Significance",
    title = "Model Coefficients Across Cell Types") +
  theme_minimal(base_size = 30) + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), legend.position = "right") + coord_flip()

#p 25x20
```


```{r}
plot_df_filtered %>% ggplot(aes(x = abs(estimate), fill = term)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + labs(title = "Density of Cell Type Effect Sizes", x = "abs(Effect Size)", y = "Density")
#L 8x5


plot_df_filtered %>% filter(p.adj<0.1) %>% ggplot(aes(x = abs(estimate), fill = term)) +
  geom_density(alpha = 0.6) + scale_fill_manual(values = c("steelblue", "darkgreen")) + theme_bw(base_size = 20) + theme(legend.position = "none", axis.title = element_blank(), plot.title = element_blank())
```


```{r}
df %>% filter(`Non-classical monocytes`>0) %>% ggplot(aes(x=urb_score.y, y=`Non-classical monocytes`)) + geom_point(color="darkgreen") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Urbanicity Score", y = "Non-Classical Monocytes")
#L 5x5 
```


```{r}
df %>%  filter(`Non-classical monocytes`>0) %>% ggplot(aes(x=Early_urban_PC1, y=`Non-classical monocytes`)) + geom_point(color="steelblue") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Early Life Score", y = "Non-Classical Monocytes")
#L 5x5 
```



```{r}
df %>% filter(`Type 1 helper T cells` > 0)  %>% ggplot(aes(x=urb_score.y, y=`Type 1 helper T cells`)) + geom_point(color="darkgreen") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Urbanicity Score", y = "Th1 cells")
#L 5x5 
```


```{r}
df %>% filter(`Type 1 helper T cells` > 0)  %>% ggplot(aes(x=Early_urban_PC1, y=`Type 1 helper T cells`)) + geom_point(color="steelblue") + theme_bw(base_size = 20) + geom_smooth(method = lm, color="black") + labs(x = "Early Life Score", y = "Th1 cells")
#L 5x5 
```


```{r}
proportions <- read_csv("subset_proportions (1).csv")
colnames(proportions)[1] = "sample_name"

proportions %<>% dplyr::select(-`Epithelial cells`, -`Intestinal macrophages`, -`Double-negative thymocytes`, -`Double-positive thymocytes`)
```

```{r}
prop_long <- proportions %>%
  tidyr::pivot_longer(
    cols = -sample_name,
    names_to = "cell_type",
    values_to = "proportion")
```

```{r}
cell_groups <- list(
  T_cells = c("CRTAM+ gamma-delta T cells", "Cycling T cells", "Early lymphoid/T lymphoid", 
              "Follicular helper T cells", "gamma-delta T cells", "MAIT cells", 
              "Memory CD4+ cytotoxic T cells", "Regulatory T cells", "T(agonist)", 
              "Tcm/Naive cytotoxic T cells", "Tcm/Naive helper T cells", 
              "Tem/Effector helper T cells", "Tem/Temra cytotoxic T cells", 
              "Tem/Trm cytotoxic T cells", "Trm cytotoxic T cells", 
              "Type 1 helper T cells", "Type 17 helper T cells", "Treg(diff)", "CD8a/a"),
  Monocytes = c("Classical monocytes", "Intermediate macrophages", "Macrophages", 
                "Monocytes", "Mono-mac", "Non-classical monocytes", "DC1", "DC2", "pDC"),
  B_cells = c("Age-associated B cells", "B cells", "Germinal center B cells", 
              "Memory B cells", "Naive B cells", "Plasma cells", 
              "Plasmablasts", "Transitional B cells"),
  NK_cells = c("CD16+ NK cells", "CD16- NK cells", "NK cells", "NKT cells"),
  Stem_Progenitor = c("HSC/MPP", "Mid erythroid", "Megakaryocytes/platelets"))
```

```{r}
group_df <- stack(cell_groups)
colnames(group_df) <- c("cell_type", "group")

prop_long <- prop_long %>%
  left_join(group_df, by = "cell_type")

prop_long$group <- as.character(prop_long$group)
prop_long$group[is.na(prop_long$group)] <- "Other"
prop_long$cell_type <- as.character(prop_long$cell_type)
prop_long$cell_type[is.na(prop_long$cell_type)] <- "Other"

```


```{r}
cell_order <- c(
  "CD8a/a", "CRTAM+ gamma-delta T cells", "Cycling T cells", "Early lymphoid/T lymphoid",
  "Follicular helper T cells", "gamma-delta T cells", "MAIT cells", 
  "Memory CD4+ cytotoxic T cells", "Regulatory T cells", "T(agonist)", 
  "Tcm/Naive cytotoxic T cells", "Tcm/Naive helper T cells", 
  "Tem/Effector helper T cells", "Tem/Temra cytotoxic T cells", 
  "Tem/Trm cytotoxic T cells", "Trm cytotoxic T cells", 
  "Type 1 helper T cells", "Type 17 helper T cells", "Treg(diff)",
  "Classical monocytes", "Intermediate macrophages", "Macrophages", 
  "Monocytes", "Mono-mac", "Non-classical monocytes", "DC1", "DC2", "pDC",
  "Age-associated B cells", "B cells", "Germinal center B cells", 
  "Memory B cells", "Naive B cells", "Plasma cells", "Plasmablasts", "Transitional B cells",
  "CD16+ NK cells", "CD16- NK cells", "NK cells", "NKT cells",
  "HSC/MPP", "Mid erythroid", "Megakaryocytes/platelets", "Other")

prop_long$cell_type <- factor(prop_long$cell_type, levels = cell_order)
```

rainbow plot
```{r}
ggplot(prop_long, aes(x = sample_name, y = proportion, fill = cell_type)) +
  geom_bar(stat = "identity") + theme_bw(base_size = 20) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  ylab("Proportion") +
  xlab("Sample") +
  guides(fill = guide_legend(ncol = 1)) 
```



add in specific coloring 
```{r}
group_colors <- c(
  T_cells = "#1f78b4",         # blue
  B_cells = "#33a02c",         # green
  NK_cells = "#e31a1c",        # red
  Monocytes = "#ff7f00",       # orange
  Stem_Progenitor = "#6a3d9a", # purple
  Other = "grey")

cell_type_colors <- prop_long %>%
  distinct(group, cell_type) %>%
  group_by(group) %>%
  summarise(cell_type_list = list(cell_type), .groups = "drop") %>%
  mutate(
    colors = lapply(seq_along(cell_type_list), function(i) {
      base_col <- group_colors[group[i]]
      n <- length(cell_type_list[[i]])
      # generate shades from lighter to base color
      colorRampPalette(c(lighten(base_col, 0.4), base_col))(n)
    })
  ) %>%
  unnest(cols = c(cell_type_list, colors)) %>%
  dplyr::rename(cell_type = cell_type_list, color = colors)

fill_colors <- setNames(cell_type_colors$color, cell_type_colors$cell_type)
```


```{r}
ggplot(prop_long, aes(x = sample_name, y = proportion, fill = cell_type)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = fill_colors) +
  theme_bw(base_size = 20) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  ylab("Proportion") +
  xlab("Sample") +
  guides(fill = guide_legend(ncol = 1))

#L 30x20
```









umap 
-------------------------------------------------------

```{r}
umap_df=readRDS("umap_plot_df_8oct25.rds")
```

```{r}
umap_df %>% ggplot(aes(x=V1, y=V2, color=cell_type_high)) + geom_point() + theme_bw(base_size = 20) + theme(legend.position = "none")
umap_df %>% ggplot(aes(x=V1, y=V2, color=cell_type_low)) + geom_point() + theme_bw(base_size = 20) + theme(legend.position = "none")
#10x10 L
```


