---
title: "making_metadata_11nov25"
author: "layla"
date: "2025-11-26"
output: html_document
---

### RNAseq data cleaning ---

```{r warning = FALSE, message = FALSE}
library(limma)
library(edgeR)
library(ggplot2)
library(corrplot)
library(googlesheets4)
library(ggrepel)
library(stringr)
library(dplyr)
gs4_deauth()
library(magrittr)
```

google sheets authorization won't work on sol cluster so i manually downloaded these sheets April 17th 2025
```{r eval=FALSE, include=FALSE}
qc1=read_sheet('https://docs.google.com/spreadsheets/d/1Boi9IQKyTjObnZzNEe015KsUNGTW2m6O2Hw7SJZiOaM/edit?gid=1456305001#gid=1456305001',sheet=1)
qc2=read_sheet('https://docs.google.com/spreadsheets/d/1Boi9IQKyTjObnZzNEe015KsUNGTW2m6O2Hw7SJZiOaM/edit?gid=1456305001#gid=1456305001',sheet=2)
qc3=read_sheet('https://docs.google.com/spreadsheets/d/1Boi9IQKyTjObnZzNEe015KsUNGTW2m6O2Hw7SJZiOaM/edit?gid=1456305001#gid=1456305001',sheet=3)
corrected=read_sheet('https://docs.google.com/spreadsheets/d/1ULwvuZuUptctv8fYn6xhRf0W3_NDm1-j2H2Nts_R0w4/edit?gid=1863156836#gid=1863156836')
```

then downloaded Audrey's fixes on Nov 11th 2025
```{r eval=FALSE, include=FALSE}
resolved = read_sheet('https://docs.google.com/spreadsheets/d/1JF7aLZbsl_JzM9DW8YbcmY55BN2oShxJTPpIVqUrFyw/edit?gid=1855266807#gid=1855266807', sheet="resolved_errors")

dont_use = read_sheet('https://docs.google.com/spreadsheets/d/1JF7aLZbsl_JzM9DW8YbcmY55BN2oShxJTPpIVqUrFyw/edit?gid=1855266807#gid=1855266807' ,sheet="unresolved_dont_use")

caution = read_sheet('https://docs.google.com/spreadsheets/d/1JF7aLZbsl_JzM9DW8YbcmY55BN2oShxJTPpIVqUrFyw/edit?gid=1855266807#gid=1855266807' ,sheet="caution_high_relatedness")
```

read in data and organize a few things
```{r}
# from Dropbox
medical=read.csv('~/vandy/early_life/redo_11nov25//medical.csv')
traditional=read.csv('~/vandy/early_life/redo_11nov25/traditional_lifestyle.csv')
personal=read.csv('~/vandy/early_life/redo_11nov25/personal_information.csv')
early=read.csv('~/vandy/early_life/redo_11nov25/early_life.csv')
urbanicity <- read.delim('~/vandy/early_life/github_17apr25/OAHeLP_location_urbanicity_scores_2025-03-08.txt')
urbanicity<-unique(urbanicity[,c('village_id','urb_score')])

# from Google
qc1=read.csv('~/vandy/early_life/github_17apr25/RNA-seq QC - round1.csv')
qc2=read.csv('~/vandy/early_life/github_17apr25/RNA-seq QC - round2.csv')
qc3=read.csv('~/vandy/early_life/github_17apr25/RNA-seq QC - round3.csv')
corrected=read.csv('~/vandy/early_life/github_17apr25/OA PBMC RNA-Seq Info - TgenForm.csv')
qc1$batch<-1
qc2$batch<-2
qc3$batch<-3
qc_all<-rbind(qc1,qc2,qc3)
# duplicate
qc_all<-qc_all[-which(qc_all$TID==613 & qc_all$Lane==2),]

# recode ethnicity
personal$ethnicity<-NA
personal$ethnicity[which(personal$ethnicity___0==1)]<-'Batek'
personal$ethnicity[which(personal$ethnicity___1==1)]<-'Jehai'
personal$ethnicity[which(personal$ethnicity___2==1)]<-'Jakun'
personal$ethnicity[which(personal$ethnicity___3==1)]<-'Mah_Meri'
personal$ethnicity[which(personal$ethnicity___4==1)]<-'Mendriq'
personal$ethnicity[which(personal$ethnicity___5==1)]<-'Orang_Kuala'
personal$ethnicity[which(personal$ethnicity___6==1)]<-'Orang_Seletar'
personal$ethnicity[which(personal$ethnicity___7==1)]<-'Semai'
personal$ethnicity[which(personal$ethnicity___8==1)]<-'Semaq_Beri'
personal$ethnicity[which(personal$ethnicity___9==1)]<-'Temiar'
personal$ethnicity[which(personal$ethnicity___10==1)]<-'Temuan'
personal$ethnicity[which(personal$ethnicity___99==1)]<-'Other'
tmp<-rowSums(personal[,18:29])
personal$ethnicity[which(tmp>1)]<-'Multiple'

# make broader groups
personal$ethnicity2<-'Other'
personal$ethnicity2[which(personal$ethnicity %in% c('Batek','Jehai','Mendriq'))]<-'Semang'
personal$ethnicity2[which(personal$ethnicity %in% c('Temuan','Jakun','Orang_Seletar','Orang_Kuala'))]<-'Proto_Malay'
personal$ethnicity2[which(personal$ethnicity %in% c('Temiar','Semai','Semaq_Beri','Mah_Meri'))]<-'Senoi'

# merge metadata
all <- merge(
  medical[, c('rid','tid','sex_medical','med_date','neutrophils_perc','lymphocytes_perc','monocytes_perc','eosinophils_perc','basophils_perc',
              'neutrophils','lymphocytes','monocytes','eosinophils','basophils',
              'neutrophils_perc_manual','lymphocytes_perc_manual','monocytes_perc_manual','eosinophils_perc_manual','interview_location_med')],
  personal[, c('rid','ethnicity','ethnicity2','other_ethnicity','date_of_birth')])
all$tid = as.character(all$tid)

all <- all %>% left_join(urbanicity, by = c("interview_location_med" = "village_id"))
all$age<-as.numeric(as.Date(all$med_date)-as.Date(all$date_of_birth))/365
```


estimate sex from the RNA-seq data as a sanity check
```{r}
# read in and filter the RNA-seq data
counts=read.delim('~/vandy/early_life/OA_countMatrix_5Mar24.txt',sep=' ')
protein_coding_genes <- readRDS("~/vandy/tcRNAseq/protein_coding_genes_29jan25.rds")
protein_coding_gene_ids <- protein_coding_genes$external_gene_name
# only keep PC genes
counts2 <- counts[rownames(counts) %in% protein_coding_gene_ids, ]

t2g <- read.delim('~/vandy/tcRNAseq/30aug24_protein_coding_gene_info.txt')
y_genes <- t2g %>% filter(chromosome_name=="Y")



# sex check
qc_tmp<-as.data.frame(names(counts2))
names(qc_tmp)[1]<-'sample_name'
qc_tmp$tot_pc_counts<-apply(counts2,2,sum)
qc_tmp$tot_counts<-apply(counts,2,sum)
qc_tmp$chrY_counts<-apply(counts[which(rownames(counts) %in% y_genes$external_gene_name),],2,sum)
qc_tmp$TID<-gsub('X', '', qc_tmp$sample_name)
qc_tmp$order<-1:dim(qc_tmp)[1]
qc_tmp$rnaseq_sex<-'female'
qc_tmp$rnaseq_sex<-NA
qc_tmp$rnaseq_sex[which(qc_tmp$chrY_counts/qc_tmp$tot_pc_counts>0.0005)]<-'male'
qc_tmp$rnaseq_sex[which(qc_tmp$chrY_counts/qc_tmp$tot_pc_counts<0.0001)]<-'female'
hist(qc_tmp$chrY_counts/qc_tmp$tot_pc_counts,breaks=50);abline(v=0.0001,col='red',lty=2,main='');abline(v=0.0005,col='blue',lty=2)
```


filtering a bit 
```{r}
qc_all<-merge(qc_tmp,qc_all,by='TID',all.x=T)
qc_all<-qc_all[order(qc_all$order),]

# remove the hemoglobin transcripts, hemoglobin contamination, and low counts
counts2=counts2[-which(rownames(counts2) %in% c('HBB','HBA1','HBA2')),]
remove<-subset(qc_all,tot_counts<1000000 | X..uniquely.mapped.reads <0.7 | X..unique.hemoglobin >10)
counts3<-counts2[,-c(which(names(counts2) %in% remove$sample_name))]

# make a TPM matrix 
tmp<-as.data.frame(rownames(counts3))
names(tmp)<-'gene'
tmp$order<-1:dim(tmp)[1]
tmp<-merge(tmp,protein_coding_genes,by.x='gene',by.y='external_gene_name')
tmp<-merge(tmp,t2g,by.x='gene',by.y='external_gene_name')
tmp$length<-tmp$end_position-tmp$start_position
len<-as.data.frame(aggregate(tmp$length~tmp$gene,FUN=mean))

counts3_v2<-counts3[which(rownames(counts3) %in% len[,1]),]

x <- counts3_v2 / len[,2]
tpm.mat <- t( t(x) * 1e6 / colSums(x) )

med_TPM<-apply(tpm.mat,1,median)
keep<-med_TPM[which(med_TPM>2)]
counts4<-counts3[which(rownames(counts3) %in% names(keep)),]

rm(counts);rm(counts2);rm(counts3)
```


merge and organize the metadata
```{r}
names<-subset(qc_all,sample_name %in% names(counts4))
names$tid2<-names$TID

# correct some issues with sample swaps
names<-merge(names,corrected[,c("CORRECTED_TID",'Patient_ID')],by.y='Patient_ID', by.x='tid2',all.x=T)

names$tid2[which(names$CORRECTED_TID!=names$TID)]<-names$CORRECTED_TID[which(names$CORRECTED_TID!=names$TID)]
names<-merge(names,all, by.x='tid2',by.y='tid',all.x=T)

miss<-which(is.na(names$urb_score))
for (i in miss){
  tmp<- subset(names,med_date %in% names$med_date[i])
  names$urb_score[i]<-median(tmp$urb_score,na.rm=T)
}

# sex check
#subset(names,sex_medical!=rnaseq_sex & batch==3)
```

add in Audrey's fixes 
```{r}
#from Audrey 
resolved=read.csv('/home/lasmall2/vandy/early_life/redo_11nov25/Orang Asli TID mismatch notes - resolved_errors (1).csv')
dont_use=read.csv('/home/lasmall2/vandy/early_life/redo_11nov25/Orang Asli TID mismatch notes - unresolved_dont_use (1).csv')
caution=read.csv('/home/lasmall2/vandy/early_life/redo_11nov25/Orang Asli TID mismatch notes - caution_high_relatedness (1).csv')

resolved %<>% dplyr::select(file_prefix, original_tid, corrected_tid)
dont_use %<>% dplyr::select(file_prefix, original_tid, corrected_tid)
caution %<>% dplyr::select(file_prefix, original_tid, corrected_tid)

resolved$ama_type = "resolved"
dont_use$ama_type = "dont_use"
caution$ama_type = "caution"

fixes = rbind(resolved, dont_use, caution)

colnames(fixes) = c("ama_file_prefix", "ama_og_tid", "ama_corrected_tid", "ama_type")
```


TID matches RNAseq tid, tid2 is previously corrected tid
ama_og_tid matches TID 
```{r}
merged_df <- merge(names, fixes, 
                   by.x = "TID", 
                   by.y = "ama_og_tid", 
                   all.x = TRUE)
```

important info 
VANP_TID356_1_PB_WBC_C1_UUUUU_L98450_227LCVLT4_AAGGAAGG --> corrected_id = 773
VANP_TID237_1_PB_WBC_C1_UUUUU_L98389_227LCVLT4_CCATCCGC --> corrected_id = 282
X282 --> also 282 
and no sample for 356


fixing so we only have 1 sample for 282 
```{r}
merged_df %>% filter(sample_name=="X282") %>% dplyr::select(tot_pc_counts, tot_counts, X..uniquely.mapped.reads)
merged_df %>% filter(ama_file_prefix=="VANP_TID237_1_PB_WBC_C1_UUUUU_L98389_227LCVLT4_CCATCCGC") %>% dplyr::select(tot_pc_counts, tot_counts, X..uniquely.mapped.reads)
```

VANP_TID237_1_PB_WBC_C1_UUUUU_L98389_227LCVLT4_CCATCCGC wins (more PC counts, basically the same mapped %)

removing things 
```{r}
merged_df_filt = merged_df %>% filter(!sample_name=="X282") #1015 
merged_df_filt = merged_df_filt %>% filter(!ama_type == "dont_use" | is.na(ama_type))
```

make FINAL tid 
```{r}
merged_df_filt$ama_corrected_tid = as.character(merged_df_filt$ama_corrected_tid)
merged_df_filt <- merged_df_filt %>%
  mutate(final_TID = if_else(is.na(ama_corrected_tid), tid2, ama_corrected_tid))
```

re-add info that's based on tid and rid 
```{r}
merged_df_filt = merged_df_filt %>% dplyr::select(final_TID, TID, tid2, sample_name, tot_pc_counts, tot_counts, chrY_counts, order, rnaseq_sex, batch, Lane, Total.reads, Uniquely.mapped.reads, X..uniquely.mapped.reads, X..multi.mapped, Hemoglobin, X..unique.hemoglobin, Sequencing_Raw_Reads.TGEN, ama_file_prefix, ama_corrected_tid, ama_type)

all_filt = all %>% filter(tid %in% merged_df_filt$final_TID)
all_filt$final_TID = as.character(all_filt$tid)
meta = left_join(merged_df_filt, all_filt, by = "final_TID")
```

jk theres more duplicates 
```{r}
dup_ids = meta[duplicated(meta$final_TID),]$final_TID
meta %>% filter(final_TID %in% dup_ids) %>% dplyr::select(final_TID, sample_name, ama_file_prefix, ama_corrected_tid, ama_type) %>% View()


meta %>% filter(final_TID %in% dup_ids) %>% dplyr::select(final_TID, sample_name, tot_pc_counts, tot_counts, X..uniquely.mapped.reads) %>% View()
```

list of sample_name and final_TID matches that are STAYING
X1001 , 1081
X1603 , 1142
X1167 , 1167
X1259 , 1259
X1471 , 1471
X1477 , 1477
X1488 , 1488
X1492 , 1492
X1377 , 1504 
X1509 , 1509
X1511 , 1511 
X1372 , 1519
X1335 , 1575
X1637 , 1637
X1677 , 1677
X304 , 363

this one is from before 
X237 , 282

this sample seems to have two RNAseq samples 
keeping X613 and removing X613.1

removing duplicates 
```{r}
meta_filtered = meta %>% filter(!sample_name %in% c("X1081", "X1142", "X1263", "X1580", "X1592", "X1284", "X1496", "X1622", "X1504", "X1545", "X1508", "X1519", "X1575", "X1173", "X1472", "X363", "X613.1"))

meta_filtered %<>% filter(!is.na(rid))
```

*here at 985 samples 


make combined variable for neut/lymph percentage
```{r}
meta_filtered$neut_perc_combine<-meta_filtered$neutrophils_perc
meta_filtered$neut_perc_combine[which(is.na(meta_filtered$neutrophils_perc))]<-meta_filtered$neutrophils_perc_manual[which(is.na(meta_filtered$neutrophils_perc))]

meta_filtered$eosinophils_perc_combine<-meta_filtered$eosinophils_perc
meta_filtered$eosinophils_perc_combine[which(is.na(meta_filtered$eosinophils_perc))]<-meta_filtered$eosinophils_perc_manual[which(is.na(meta_filtered$eosinophils_perc))]

meta_filtered$monocytes_perc_combine<-meta_filtered$monocytes_perc
meta_filtered$monocytes_perc_combine[which(is.na(meta_filtered$monocytes_perc))]<-meta_filtered$monocytes_perc_manual[which(is.na(meta_filtered$monocytes_perc))]

meta_filtered$lympho_perc_combine<-meta_filtered$lymphocytes_perc
meta_filtered$lympho_perc_combine[which(is.na(meta_filtered$lymphocytes_perc))]<-meta_filtered$lymphocytes_perc_manual[which(is.na(meta_filtered$lymphocytes_perc))]

meta_filtered$lympho_perc_combine[which(is.na(meta_filtered$lympho_perc_combine))]<-mean(meta_filtered$lympho_perc_combine,na.rm=T)
meta_filtered$neut_perc_combine[which(is.na(meta_filtered$neut_perc_combine))]<-mean(meta_filtered$neut_perc_combine,na.rm=T)
meta_filtered$eosinophils_perc_combine[which(is.na(meta_filtered$eosinophils_perc_combine))]<-mean(meta_filtered$eosinophils_perc_combine,na.rm=T)
meta_filtered$monocytes_perc_combine[which(is.na(meta_filtered$monocytes_perc_combine))]<-mean(meta_filtered$monocytes_perc_combine,na.rm=T)

pca<-prcomp(meta_filtered[,c('monocytes_perc_combine','neut_perc_combine','eosinophils_perc_combine','lympho_perc_combine')],scale=T)
meta_filtered$PC1_cell<-pca$x[,1]
meta_filtered$PC2_cell<-pca$x[,2]
cor.test(meta_filtered$lympho_perc_combine,meta_filtered$PC1_cell)

meta_filtered<-meta_filtered[order(meta_filtered$order),]
```


```{r}
#write.table(meta_filtered, "~/vandy/early_life/redo_11nov25/meta_cell_12nov25.txt")
```




need to subset counts data to match samples we're keeping before normalizing 
```{r}
counts5 <- counts4 %>% select(any_of(meta_filtered$sample_name))
```


voom norm and save 
```{r}
dge <- DGEList(counts=counts5)
dge <- calcNormFactors(dge)

#v <- voom(dge, plot = FALSE)
#saveRDS(v,file='~/vandy/early_life/redo_11nov25/12nov25_voom_normalized.rds')


#v <- voomWithQualityWeights(dge, plot = FALSE)
#saveRDS(v,file='~/vandy/early_life/redo_11nov25/12nov25_voomWithQualityWeights_normalized.rds') aka exp.rds !!!!
```


----------------------------------------------

pca and covariate detection 
```{r}
rm(list = ls())
```

```{r}
meta=read.delim('~/vandy/early_life/redo_11nov25/meta_cell_12nov25_985samps.txt', sep = ' ')
v=readRDS('~/vandy/early_life/redo_11nov25/12nov25_voomWithQualityWeights_normalized.rds')
```

run pca and plot
```{r}
pca<-prcomp(t(v$E))
plot(pca$x[,1],pca$x[,2])
```

join GE pcs w/ metadata 
```{r}
pcs <- data.frame(pca$x[,1:2])
pcs$sample_name <- rownames(pcs)
meta <- left_join(meta, pcs, by = "sample_name")
```

```{r}
meta %>% ggplot(aes(x = PC1, y = PC2, label = sample_name)) + geom_text(size = 3) + theme_bw()
```

```{r}
meta_numeric <- meta
meta_numeric$sex_numeric = ifelse(meta_numeric$rnaseq_sex=="female", 1,0)
meta_numeric %<>% dplyr::select(where(is.numeric))
meta_numeric %<>% select(-tid2, -TID, -tid,-Sequencing_Raw_Reads.TGEN, -order, -interview_location_med, -final_TID, -neutrophils_perc, -lymphocytes_perc, -monocytes_perc, -eosinophils_perc, -basophils, -neutrophils_perc_manual, -lymphocytes_perc_manual, -monocytes_perc_manual, -eosinophils_perc_manual, -ama_corrected_tid)
meta_numeric %<>% na.omit(dat_numeric)

pheatmap::pheatmap(cor(meta_numeric), display_numbers = TRUE, number_color = "black")
```

```{r}
meta %>% ggplot(aes(x=PC1, y=PC2, color=Total.reads, shape=as.factor(batch))) + geom_point(size=3) + theme_classic() 
```


```{r}
meta %>% ggplot(aes(x=Total.reads, fill=as.factor(batch))) + geom_histogram() + theme_classic()

meta %>% ggplot(aes(x=tot_pc_counts, fill=as.factor(batch))) + geom_histogram() + theme_classic()

meta %>% ggplot(aes(x=X..uniquely.mapped.reads, fill=as.factor(batch))) + geom_histogram(stat="count") + theme_classic()
```

--------------------------------------------------------

### early life score test ---

re-making early life score to see if any new ppl appear 

```{r}
rm(list = ls())
```

```{r warning = FALSE, message = FALSE}
library(tidyverse)
library(factoextra)
library(FactoMineR)
library(factoextra)
#library(exactextractr)
library(geosphere)
library(mice)
library(psych)
library(ggplot2)
library(ggmap)
library(ggrepel)
library(dplyr)
library(data.table)
library(ggrepel)
library(ggforce)
library(scales)
#library(preprocessCore)
library(nlWaldTest)
#library(nationalparkcolors)
#pal <- park_palette("Badlands")
library(egg)
```

read in the data, run Tom's approved cleaning code 
```{r, results = 'hide', include=FALSE} 
Medical=read.csv('~/vandy/early_life/redo_11nov25/medical.csv')
Traditional=read.csv('~/vandy/early_life/redo_11nov25/traditional_lifestyle.csv')
Personal_Information=read.csv('~/vandy/early_life/redo_11nov25/personal_information.csv')
Early_Life=read.csv('~/vandy/early_life/redo_11nov25/early_life.csv')
Visit_Register <- read.csv('~/vandy/early_life/redo_11nov25/visit_register.csv')
Village_Register <- read.csv('~/vandy/early_life/redo_11nov25/oa_village_register.csv')
Urbanicity <- read.delim('~/vandy/early_life/redo_11nov25/OAHeLP_location_urbanicity_scores_2025-07-11.txt')
```


```{r}
OA_data <- left_join(Medical, Personal_Information, by = "rid", suffix = c(".medical", ".personal"))
table(OA_data$vid %in% Traditional$vid)  # 1202 have interviews, 133 missing.
OA_data <- left_join(OA_data, Traditional, by = c("rid","vid"))
table(OA_data$vid %in% Early_Life$vid)  # 1209 have interviews, 126 missing. 
OA_data <- left_join(OA_data, Early_Life, by = c("rid","vid"))
table(OA_data$interview_location_med %in% Village_Register$village_id)  # 100% matches. 
OA_data$lat <- Village_Register$lat[match(OA_data$interview_location_med, Village_Register$village_id)]
OA_data$long <- Village_Register$long[match(OA_data$interview_location_med, Village_Register$village_id)]
names(OA_data)[which(names(OA_data) == "interview_location_med")] <- "village_id"
```

code ethnolinguistic groups
```{r}
OA_data$ethnicity<-NA
OA_data$ethnicity[which(OA_data$ethnicity___0==1)]<-'Batek'
OA_data$ethnicity[which(OA_data$ethnicity___1==1)]<-'Jehai'
OA_data$ethnicity[which(OA_data$ethnicity___2==1)]<-'Jakun'
OA_data$ethnicity[which(OA_data$ethnicity___3==1)]<-'Mah_Meri'
OA_data$ethnicity[which(OA_data$ethnicity___4==1)]<-'Mendriq'
OA_data$ethnicity[which(OA_data$ethnicity___5==1)]<-'Orang_Kuala'
OA_data$ethnicity[which(OA_data$ethnicity___6==1)]<-'Orang_Seletar'
OA_data$ethnicity[which(OA_data$ethnicity___7==1)]<-'Semai'
OA_data$ethnicity[which(OA_data$ethnicity___8==1)]<-'Semaq_Beri'
OA_data$ethnicity[which(OA_data$ethnicity___9==1)]<-'Temiar'
OA_data$ethnicity[which(OA_data$ethnicity___10==1)]<-'Temuan'
OA_data$ethnicity[which(OA_data$ethnicity___99==1)]<-'Other'
OA_data$ethnicity[which(OA_data$other_ethnicity %in% c("kensiu", "Kensiu", "Kensiu, melayu (father)"))]<-'Kensiu'
OA_data$ethnicity[which(OA_data$other_ethnicity %in% c("semelai", "Semelai"))]<-'Semelai'
OA_data$ethnicity[which(is.na(OA_data$ethnicity) & OA_data$interview_location_med == 25)]<-'Jakun'  # this is Kg Petoh, so almost certainly jakun

sum(is.na(OA_data$ethnicity))  # 13 missing, other categories like melayu

# Calculate the row sums for the specific columns
tmp <- rowSums(OA_data[, c("ethnicity___0", "ethnicity___1", "ethnicity___2", "ethnicity___3", 
                                 "ethnicity___4", "ethnicity___5", "ethnicity___6", "ethnicity___7", 
                                 "ethnicity___8", "ethnicity___9", "ethnicity___10", "ethnicity___99")])

# Assign 'Multiple' to ethnicity if more than one ethnicity is indicated
OA_data$ethnicity[which(tmp > 1)] <- 'Multiple'

# make broader groups
OA_data$ethnicity2<-'Other'
OA_data$ethnicity2[which(OA_data$ethnicity %in% c('Batek','Jehai','Mendriq', "Kensiu"))]<-'Semang'
OA_data$ethnicity2[which(OA_data$ethnicity %in% c('Temuan','Jakun','Orang_Seletar','Orang_Kuala','Semelai'))]<-'Proto_Malay'
OA_data$ethnicity2[which(OA_data$ethnicity %in% c('Temiar','Semai','Semaq_Beri','Mah_Meri'))]<-'Senoi'

# Calculate age
OA_data <- OA_data %>%
  mutate(age = as.numeric(ymd(med_date) - ymd(date_of_birth)) / 365.25)
```


specific corrections from Rob
```{r}
OA_data$height_hip[which(OA_data$tid == "1269")] <- 85
OA_data$height_hip[which(OA_data$tid == "929")] <- 98
OA_data$height_hip[which(OA_data$tid == "1011")] <- 97

OA_data$height_hip[which(OA_data$tid == "1233")] <- 100
OA_data$height_knee[which(OA_data$tid == "1233")] <- 51
OA_data$waist_circum[which(OA_data$tid == "1233")] <- 110
OA_data$hip_circum[which(OA_data$tid == "1233")] <- 114

# Define the correction list
waist_correction <- c("1636", 	
                      "1491",
                      "1145")
tar <- which(OA_data$tid %in% waist_correction)
OA_data$waist_circum[tar] <- OA_data$waist_circum[tar] *2.54

# waist outliers entered wrong
OA_data$waist_circum[which(OA_data$tid == 1602)] <- 74.2

# Define the correction list for hip circumference
hip_correction <- c("1636")  # Added missing rid values

tar <- which(OA_data$tid %in% hip_correction)
OA_data$hip_circum[tar] <- OA_data$hip_circum[tar] *2.54

# waist outliers entered wrong
OA_data$hip_circum[which(OA_data$tid == 1504)] <- 73
OA_data$hip_circum[which(OA_data$tid == 1486)] <- NA #implausible value
OA_data$hip_circum[which(OA_data$tid == 1258)] <- 80.5 
OA_data$hip_circum[which(OA_data$tid == 1235)] <- 101 # this value was crossed out but replaced by another value that doesn't make sense 

# Fix problematic height_knee observations
hist(OA_data$height_knee)
filter(OA_data, height_knee < 35 | height_knee > 57) %>% dplyr::select(rid, med_date, village_id, sex_medical:temp, tid)

# implausible but unfixable. Others seem real.
OA_data$height_knee[which(OA_data$tid == 780)] <- NA
```

calculate BMI/body comp metrics now that these variables are cleaned - then look at who may be outside the NHANES range
```{r}
OA_data <- OA_data %>%
  mutate(
    BMI = weight / ((height_standing/100) * (height_standing/100)),
    waist_height = waist_circum / height_standing,
    waist_hip = waist_circum/hip_circum,
    BRI = 364.2 - 365.5*sqrt(1-(waist_circum/(2*pi))^2/(0.5*height_standing)^2)
  )

OA_data[which(OA_data$waist_hip > 1.35), "hip_circum"] <- NA
OA_data[which(OA_data$waist_hip > 1.35), "waist_circum"] <- NA
OA_data[which(OA_data$waist_hip > 1.35), "waist_hip"] <- NA

# And I'm going to use .65 as the lower end for now (which is NHANES)
OA_data[which(OA_data$waist_hip < 0.65), "hip_circum"] <- NA
OA_data[which(OA_data$waist_hip < 0.65), "waist_circum"] <- NA
OA_data[which(OA_data$waist_hip < 0.65), "waist_hip"] <- NA

# Make triglycerides numeric
OA_data[which(OA_data$triglyc=="<0.56"),]$triglyc <- "0.56"
OA_data[which(OA_data$triglyc==">5.65"),]$triglyc <- "5.65"
OA_data[grep("<",OA_data$total_chol),]$total_chol <- "2.59"
OA_data[grep(">",OA_data$total_chol),]$total_chol <- "10.36"
OA_data[grep("<",OA_data$hdl),]$hdl <- "0.52"
OA_data[grep("<",OA_data$hba1c),]$hba1c <- "4.0"
OA_data[,c("hdl","total_chol","triglyc","hba1c","ldl")] <- sapply(OA_data[,c("hdl","total_chol","triglyc","hba1c","ldl")],function(x) as.numeric(x))

# fixes
OA_data$ldl[which(OA_data$tid == "1033")] <- 0.94

# Transform metric cholesterol data to imperial system 
# https://heartcare.sydney/cholesterol-unit-conversion/#:~:text=To%20convert%20from%20mmol%2FL,in%20mmol%2FL%20by%2038.67.

#To convert Total cholesterol, LDL and HDL from mg/dL to mmol/L, divide the value in mg/dL by 38.67. To convert from mmol/L to mg/dL, multiply the value in mmol/L by 38.67.
OA_data$total_chol <- OA_data$total_chol*38.67 # is currently in mmol/L, change to imperial mg/dL
OA_data$hdl <- OA_data$hdl*38.67
OA_data$ldl <- OA_data$ldl*38.67

#To convert Triglyceride values from mg/dL to mmol/L, divide the value in mg/dL by 88.57. To convert from mmol/L to mg/dL, multiply the value in mmol/L by 88.57.
OA_data$triglyc <- OA_data$triglyc*88.57

# Make LDL:HDL
OA_data$ldl_hdl <- OA_data$ldl/OA_data$hdl

# Remove some low values 
OA_data[which(OA_data$ldl_hdl < 0.5),]$ldl_hdl <- NA
OA_data[which(OA_data$ldl < 25),]$ldl <- NA

# Make non-hdl
OA_data$non_hdl = OA_data$total_chol - OA_data$hdl

# OUTLIER FIXES from AJL
# removing remaining outliers - many height outliers from the same date - 2020-03-04
OA_data$BRI[which(OA_data$BRI>20)]<-NA
OA_data$height_standing[which(OA_data$height_standing<130 | OA_data$height_standing>180)]<-NA

# create some cutoff/binary variables
OA_data$obese<-ifelse(OA_data$BMI>30,1,0)
OA_data$hypertension<-ifelse(OA_data$sys_bp2>129 | OA_data$dias_bp2>79 ,1,0)
OA_data$t2d<-ifelse(OA_data$glucose>199 | OA_data$hba1c>6.4 ,1,0)
```

run Rob's code to get early life PCs - made a few edits to variables
```{r, results = 'hide }
# Can you use a motorized vehicle on land to reach their village?
OA_data <- OA_data %>%
  mutate(early_vill_access_road = ifelse(  early_vill_access_by___car == 1 | 
                                              early_vill_access_by___truck == 1, 
                                            1, 0))

OA_data <- OA_data %>%
  mutate(birth_place_hosp = ifelse(birth_place_type == 0, 1, 0))
OA_data$index<-1:dim(OA_data)[1]

# for people with one parent, add data from the other parent
dad_missing<-subset(OA_data,is.na(early_father_occupation___crop & early_father_occupation___subsist_crop & early_father_occupation___forage & early_father_occupation___wage) & !is.na(early_mother_occupation___crop & early_mother_occupation___subsist_crop & early_mother_occupation___forage & early_mother_occupation___wage))
mom_missing<-subset(OA_data,!is.na(early_father_occupation___crop & early_father_occupation___subsist_crop & early_father_occupation___forage & early_father_occupation___wage) & is.na(early_mother_occupation___crop & early_mother_occupation___subsist_crop & early_mother_occupation___forage & early_mother_occupation___wage))

# if dad is missing, replace with mom's info
OA_data$early_father_occupation___crop[dad_missing$index]<-OA_data$early_mother_occupation___crop[dad_missing$index]
OA_data$early_father_occupation___subsist_crop[dad_missing$index]<-OA_data$early_mother_occupation___subsist_crop[dad_missing$index]
OA_data$early_father_occupation___forage[dad_missing$index]<-OA_data$early_mother_occupation___forage[dad_missing$index]
OA_data$early_father_occupation___wage[dad_missing$index]<-OA_data$early_mother_occupation___wage[dad_missing$index]

# if mom is missing, replace with dad's info
OA_data$early_mother_occupation___crop[mom_missing$index]<-OA_data$early_father_occupation___crop[mom_missing$index]
OA_data$early_mother_occupation___subsist_crop[mom_missing$index]<-OA_data$early_father_occupation___subsist_crop[mom_missing$index]
OA_data$early_mother_occupation___forage[mom_missing$index]<-OA_data$early_father_occupation___forage[mom_missing$index]
OA_data$early_mother_occupation___wage[mom_missing$index]<-OA_data$early_father_occupation___wage[mom_missing$index]

# Variable Groups
parent_occupation <-   c("early_father_occupation___crop",
                         "early_father_occupation___subsist_crop",
                         "early_father_occupation___forage",
                         "early_father_occupation___wage",
                         "early_mother_occupation___crop",
                         "early_mother_occupation___subsist_crop",
                         "early_mother_occupation___forage",
                         "early_mother_occupation___wage")

hh_item <- c( "early_hh_item___moto",
              "early_hh_item___bicycle",
              "early_hh_item___generator",
              "early_hh_item___blowpipe",
              "early_hh_item___machete",
              "early_hh_item___flip_phone",
              "early_hh_item___smart_phone",
              "early_hh_item___chainsaw",
              "early_hh_item___weedwacker",
              "early_hh_item___car",
              "early_hh_item___tv",
              "early_hh_item___gas_stove")

diet <- c("early_store",
          "early_oil_sugar",
          "early_wild_meat")

early_misc <- c("early_school",
                "birth_place_hosp",
                # AJL added
                "early_vill_access_road",
                "early_clean_water"  )

housing <- c( "early_house_type___traditional",
               "early_house_type___stone",
               "early_house_type___wood")

# Define the variable groups and their colors
variable_groups <- list(
  parent_occupation = parent_occupation,
  hh_item = hh_item,
  diet = diet,
  early_misc = early_misc,
  housing = housing)

group_colors <- c("#5495CFFF", "#F5AF4DFF", "#DB4743FF", "#7C873EFF", "#FEF4D5FF")
group_labels <- c("Parent Occupation", "Household Items", "Diet", "Miscellaneous Environment", "Housing")

# List of variables to check for completeness
variables_to_check <- c(parent_occupation, diet, housing, early_misc, hh_item)

dim(OA_data)
# which variables have missing data
apply( OA_data[,variables_to_check] ,2,function(x) length(which(is.na(x))) )
# how many variables are missing for a given person
x<-(apply( OA_data[,variables_to_check] ,1,function(x) length(which(is.na(x))) ))
table(x)

# Remove observations with missing data for variables_to_check (allow 1 missing data point)
OA_data_complete <- OA_data[which(x<2), variables_to_check]
dim(OA_data_complete)
OA_data_complete_imputed <- complete(mice(OA_data_complete, method ="lasso.norm"))

# Perform PCA
pca_result <- prcomp(OA_data_complete_imputed, center = TRUE, scale. = TRUE)

# Display PCA results
s <- summary(pca_result)
dat <- data.frame(
  component = factor(1:length(s$sdev), labels=paste0("PC", 1:length(s$sdev))),
  var_explained = s$sdev^2/sum(s$sdev^2)
)
ggplot(dat[1:10,], aes(y=var_explained)) + 
  geom_line(aes(x=component, group=1)) + 
  geom_point(aes(x=component)) + 
  theme_bw(13) + 
  ylab('Percent variance explained')

# Extract the PC scores
OA_data$Early_urban_PC1<-NA
OA_data$Early_urban_PC1[which(x<2)]<-pca_result$x[,1]
OA_data$Early_urban_PC2<-NA
OA_data$Early_urban_PC2[which(x<2)]<-pca_result$x[,2]

# add current urbanicity
Urbanicity <-unique(Urbanicity[, c("village_id","urb_score")])
Urbanicity <-Urbanicity[complete.cases(Urbanicity),]
OA_data2 <- left_join(OA_data, Urbanicity[, c("village_id","urb_score")], by = "village_id")

# keep these data
#write.table(OA_data2,'~/vandy/early_life/redo_11nov25/12nov25_OA_merged.txt',quote=F,row.names=F,sep='\t')
```

make the barplot of loadings for PC1
```{r}
loadings_df <- as.data.frame(pca_result$rotation[,1])
names(loadings_df)<-'loadings'
loadings_df$variable<-rownames(loadings_df)
loadings_df$group<-NA
loadings_df$group[which(loadings_df$variable %in% diet)]<-'diet'
loadings_df$group[which(loadings_df$variable %in% parent_occupation)]<-'parent occupation'
loadings_df$group[which(loadings_df$variable %in% early_misc)]<-'misc'
loadings_df$group[which(loadings_df$variable %in% hh_item)]<-'household itens'
loadings_df$group[which(loadings_df$variable %in% housing)]<-'housing'

PC1_bar<-ggplot(loadings_df, aes(x = reorder(variable, -loadings), y = loadings, fill = group)) +
  geom_bar(stat = "identity", color="black") +
  coord_flip() +
  labs(title = "PC1 Loadings", x = "Early life variables", y = "Loading", fill="Variable Groups") +
  theme_minimal() +
     guides(alpha = "none") + 
  scale_alpha_manual(values= c(0.5, 1)) +
  scale_x_discrete(labels = function(x) sub("\\..*", "", x)) +
  theme(plot.title=element_text(hjust=0.5), plot.subtitle=element_text( hjust=0.5)) +
  scale_fill_manual(values=c(group_colors[1], "grey", group_colors[2], group_colors[3], group_colors[4])); PC1_bar


```


no new data added so we'll go with the originally calculated scores 

---------------------------------------------------

### let's put it all together!! ---

```{r}
rm(list = ls())
```

```{r}
meta=read.delim('~/vandy/early_life/redo_11nov25/meta_cell_12nov25_985samps.txt', sep = ' ')
v=readRDS('~/vandy/early_life/redo_11nov25/12nov25_voomWithQualityWeights_normalized.rds')$E
dat=read.delim('~/vandy/early_life/redo_11nov25/18apr25_OA_merged.txt')
```


making h_sol 
```{r}
vars_need=dat %>% dplyr::select(rid, wage_past_month, where_poop___toilet, where_poop___river_lake, where_poop___forest, where_poop___latrine, where_poop___field ,electricity_resid, electricity_resid_source___power_lines, hh_item___tv, hh_item___smart_phone, highest_education_stage, vill_access_by___car, vill_access_by___truck, vill_access_by___boat, vill_access_by___none, vill_access_by___moto, house_type, water_source___filter, water_source___other, water_source___river_lake, water_source___well)

skimr::skim(vars_need)

vars_need$NA_count <- rowSums(is.na(vars_need))

hist(vars_need$NA_count)

columns_of_interest = colnames(vars_need)[2:22]

#df_filtered <- dat %>% filter(if_all(all_of(columns_of_interest), ~ !is.na(.x))) #1177
df_filtered <- dat %>% filter(rowSums(is.na(across(all_of(columns_of_interest)))) < 10) #1193

df_filtered$water_source___gov_tap <- 0
df_filtered[which(df_filtered$water_source___filter==0 & df_filtered$water_source___river_lake==0 & df_filtered$water_source___well==0),]$water_source___gov_tap <- 1

df_filtered <- df_filtered %>% 
  mutate(
    highest_where_poop = dplyr::case_when(
      where_poop___toilet == 1 ~ 1,
      where_poop___latrine == 1 ~ 0.5,
      where_poop___field == 1 ~ 0,
      where_poop___forest == 1 ~ 0,
      where_poop___river_lake == 1 ~ 0,
      TRUE ~ NA_real_),
    house_type_score = dplyr::case_when(
      house_type == "concrete" ~ 1,
      house_type == "wood" ~ 1,
      house_type == "traditional" ~ 0,
      TRUE ~ NA_real_),
    water_source_score = dplyr::case_when(
      water_source___filter == 1 ~ 1,
      water_source___well == 1 ~ 1,
      water_source___gov_tap == 1 ~ 1,
      water_source___river_lake == 1 ~ 0,
      TRUE ~ NA_real_))


##scaled 
df_filtered <- df_filtered %>% 
  rowwise() %>%
  mutate(h_sol = (sum(c(house_type_score, house_type_score,
                        water_source_score, highest_where_poop,
                        electricity_resid), na.rm = TRUE))/
           (sum(!is.na(c(house_type_score, house_type_score,
                         water_source_score, highest_where_poop,
                         electricity_resid))))) %>%
  ungroup()


##unscaled 
df_filtered <- df_filtered %>% 
    rowwise() %>%
    mutate(h_sol2 = (sum(c(house_type_score, house_type_score,
                          water_source_score, highest_where_poop,
                          electricity_resid), na.rm = TRUE))) %>% ungroup()

hist(df_filtered$h_sol)
hist(df_filtered$h_sol2)
```


add everything together 
```{r}
df_join = df_filtered %>% select(rid, h_sol, h_sol2)

df_join2 = left_join(dat, df_join, by="rid")

df_join2 %<>% dplyr::select(rid, h_sol, h_sol2, Early_urban_PC1, Early_urban_PC2)

df_join2 %<>% filter(rid %in% meta$rid)

#duplicated ids are just NAs
df_join2_unique <- df_join2 %>% arrange(rid, is.na(Early_urban_PC1)) %>% distinct(rid, .keep_all = TRUE)

dat2 = left_join(meta, df_join2_unique, by="rid")
```


add in updated urbanicity scores 
location 12 (Simoi Lama) needs to be given same score as location 2 (Dayok)
```{r}
urbanicity <- read.delim('~/vandy/early_life/redo_11nov25/OAHeLP_location_urbanicity_scores_2025-07-11.txt')
new_row <- data.frame(Date = "2022-06-13", village_id = 12, Name.of.village = "Simoi Lama", community_score=20.450, urb_score=24.620)
urbanicity2 <- rbind(urbanicity, new_row)
urbanicity2 %<>% dplyr::select(-Date)

names(dat2)[which(names(dat2) == "interview_location_med")] <- "village_id"
dat2 %<>% dplyr::select(-urb_score)

dat3 = left_join(dat2, urbanicity2, by="village_id")
```

removing samples with incomplete data 
```{r}
dat4 = dat3 %>% filter(!is.na(Early_urban_PC1)) #946
dim(subset(dat4,!is.na(age) & !is.na(sex_medical))) #946
dat4 = subset(dat4, age>=17.5) #929
dim(subset(dat4, !sex_medical == rnaseq_sex)) #3 keeping for now 
cat("final df size:", dim(dat4), "\n")
```

regress out batch effects
```{r}
dat4$age_scale<-as.numeric(scale(dat4$age))
dat4$urb_scale<-as.numeric(scale(dat4$urb_score))
dat4$early_scale<-as.numeric(scale(dat4$Early_urban_PC1))

v2=v[,dat4$sample_name]
identical(dat4$sample_name,colnames(v2))

design = model.matrix(~as.factor(dat4$batch) + dat4$Uniquely.mapped.reads + dat4$monocytes_perc_combine + dat4$lympho_perc_combine)
fit <-lmFit(v2,design)
fit <- eBayes(fit)
exp<-residuals.MArrayLM(object=fit, v2)
```


match with kinship data 
```{r}
OApbmcKinshipdata <- readRDS("~/vandy/early_life/redo_11nov25/OApbmcKinshipdata.rds")
colnames(OApbmcKinshipdata) <- paste0("X", colnames(OApbmcKinshipdata))
rownames(OApbmcKinshipdata) <- paste0("X", rownames(OApbmcKinshipdata))
dat4 %<>% filter(sample_name %in% colnames(OApbmcKinshipdata)) #925
K = OApbmcKinshipdata[dat4$sample_name,dat4$sample_name]
```


also adding in genomic pcs 
```{r}
eigenvecOApbmcs <- read_csv("~/vandy/early_life/redo_11nov25/eigenvecOApbmcs.csv")
eigenvecOApbmcs %<>% dplyr::select(sample_name, PC1, PC2)
colnames(eigenvecOApbmcs) <- c("sample_name", "genomic_PC1", "genomic_PC2")
dat4 = left_join(dat4, eigenvecOApbmcs, by="sample_name")
```


making sample map 
```{r}
sample_map = dat4 %>% dplyr::select(final_TID, rid, sample_name)
colnames(sample_map) = c("tid", "rid", "sample_name")
```


filter exp 
```{r}
exp2 = exp[,dat4$sample_name]
```


saving important stuff here!!!
```{r}
# saveRDS(K,file='~/vandy/early_life/redo_11nov25/K_13nov25.rds')
# saveRDS(dat4,file='~/vandy/early_life/redo_11nov25/meta_data_13nov25_pre_group_fixes.rds')
# saveRDS(exp2,file='~/vandy/early_life/redo_11nov25/exp_13nov25_batch_effects_removed.rds')
# saveRDS(sample_map,file='~/vandy/early_life/redo_11nov25/sample_map_13nov25.rds')
```


### re-coding ethnicity ---
------------------------------------------

```{r}
rm(list = ls())
```


```{r}
meta <- readRDS("~/vandy/early_life/redo_11nov25/meta_data_13nov25_pre_group_fixes.rds")
Personal_Information=read.csv('~/vandy/early_life/redo_11nov25/personal_information.csv')
```



```{r}
tmp = Personal_Information %>% select(rid, ethnicity___0, ethnicity___1, ethnicity___2, ethnicity___3, ethnicity___4, ethnicity___5, ethnicity___6, ethnicity___7, ethnicity___8, ethnicity___9, ethnicity___10, ethnicity___99) 

meta2 = left_join(meta, tmp, by="rid")

meta2$sum = rowSums(meta2[, c("ethnicity___0", "ethnicity___1", "ethnicity___2", "ethnicity___3", 
                             "ethnicity___4", "ethnicity___5", "ethnicity___6", "ethnicity___7", 
                             "ethnicity___8", "ethnicity___9", "ethnicity___10", "ethnicity___99")])

meta2$ethnicity___0[which(meta2$ethnicity___0==1)]<-'Batek'
meta2$ethnicity___1[which(meta2$ethnicity___1==1)]<-'Jehai'
meta2$ethnicity___2[which(meta2$ethnicity___2==1)]<-'Jakun'
meta2$ethnicity___3[which(meta2$ethnicity___3==1)]<-'Mah_Meri'
meta2$ethnicity___4[which(meta2$ethnicity___4==1)]<-'Mendriq'
meta2$ethnicity___5[which(meta2$ethnicity___5==1)]<-'Orang_Kuala'
meta2$ethnicity___6[which(meta2$ethnicity___6==1)]<-'Orang_Seletar'
meta2$ethnicity___7[which(meta2$ethnicity___7==1)]<-'Semai'
meta2$ethnicity___8[which(meta2$ethnicity___8==1)]<-'Semaq_Beri'
meta2$ethnicity___9[which(meta2$ethnicity___9==1)]<-'Temiar'
meta2$ethnicity___10[which(meta2$ethnicity___10==1)]<-'Temuan'

meta2 %>% filter(sum>1) %>% select(rid, ethnicity___0, ethnicity___1, ethnicity___2, ethnicity___3, ethnicity___4, ethnicity___5, ethnicity___6, ethnicity___7, ethnicity___8, ethnicity___9, ethnicity___10, ethnicity___99, other_ethnicity)

meta2$other_ethnicity=stringr::str_to_title(meta2$other_ethnicity)

meta2$ethnicity2[which(meta2$other_ethnicity=="Semelai")]<-"Proto_Malay" 
meta2$ethnicity2[which(meta2$rid=="56")]<-"Senoi"
meta2$ethnicity2[which(meta2$rid=="D7HXN")]<-"Proto_Malay"
meta2$ethnicity2[which(meta2$other_ethnicity=="Kensiu")]<-"Semang" 

table(meta2$ethnicity2)
```


```{r}
#group fixes 
meta2$ethnicity_groups<-'Other/Mixed'
meta2$ethnicity_groups[which(meta2$ethnicity %in% c('Batek','Jehai','Mendriq', "Kensiu", "Lanoh", "Kintaq"))]<-'Semang'
meta2$ethnicity_groups[which(meta2$ethnicity %in% c('Temuan','Jakun','OrangSeletar','OrangKuala','Semelai'))]<-'Proto_Malay'
meta2$ethnicity_groups[which(meta2$ethnicity %in% c('Temiar','Semai','SemaqBeri','MahMeri', 'Semelai'))]<-'Senoi'

table(meta2$ethnicity_groups)

meta2$ethnicity_groups[which(meta2$other_ethnicity=="Semelai")]<-"Proto_Malay" 
meta2$ethnicity_groups[which(meta2$rid=="56")]<-"Senoi"
meta2$ethnicity_groups[which(meta2$rid=="D7HXN")]<-"Proto_Malay"
meta2$ethnicity_groups[which(meta2$other_ethnicity=="Kensiu")]<-"Semang" 
meta2$ethnicity_groups[which(meta2$ethnicity=="Semaq_Beri")]<-"Senoi"
meta2$ethnicity_groups[which(meta2$ethnicity=="Mah_Meri")]<-"Senoi"

table(meta2$ethnicity_groups)
```

SAVING FINAL METADATA HERE
```{r}
# saveRDS(meta2,file='~/vandy/early_life/redo_11nov25/meta_data_13nov25_AFTER_group_fixes.rds')
```



### let's look at this data ---
-----------------------------------------------------

```{r}
rm(list = ls())
```

```{r}
meta <- readRDS("~/vandy/early_life/redo_11nov25/meta_data_13nov25_AFTER_group_fixes.rds")
```

```{r}
table(meta$ethnicity_groups)
```


```{r}
meta %>% ggplot(aes(x=ethnicity_groups, y=urb_score)) + geom_boxplot() + theme_bw(base_size = 15)
meta %>% ggplot(aes(x=ethnicity_groups, y=Early_urban_PC1)) + geom_boxplot() + theme_bw(base_size = 15)
meta %>% ggplot(aes(x=Early_urban_PC1, y=urb_score)) + geom_point() + theme_bw(base_size = 15)
```

```{r}
meta %>% ggplot(aes(x=urb_score, fill=ethnicity_groups)) + geom_histogram() + theme_bw(base_size = 15)
```

```{r}
meta %>% ggplot(aes(x=Early_urban_PC1, fill=ethnicity_groups)) + geom_histogram() + theme_bw(base_size = 15)
```


```{r}
meta %>% ggplot(aes(x=age, fill=sex_medical)) + geom_density(alpha=0.8) + theme_bw(base_size = 20) + scale_fill_manual(values = c("#17154FFF","#D95E31FF")) + labs(x = "Age", y="Density", fill="Sex")
```



```{r}
meta %>% ggplot(aes(x=genomic_PC1, y=genomic_PC2, color=ethnicity_groups)) + geom_point() + theme_bw(base_size = 15)
meta %>% ggplot(aes(x=genomic_PC1, y=genomic_PC2, color=urb_score)) + geom_point() + theme_bw(base_size = 15)
meta %>% ggplot(aes(x=genomic_PC1, y=genomic_PC2, color=Early_urban_PC1)) + geom_point() + theme_bw(base_size = 15)
```















### making larger exp (not filtering for EL) ---
```{r}
rm(list = ls())
```

```{r}
meta=read.delim('~/vandy/early_life/redo_11nov25/meta_cell_12nov25_985samps.txt', sep = ' ')
v=readRDS('~/vandy/early_life/redo_11nov25/12nov25_voomWithQualityWeights_normalized.rds')$E
dat=read.delim('~/vandy/early_life/redo_11nov25/18apr25_OA_merged.txt')
```


making h_sol 
```{r}
vars_need=dat %>% dplyr::select(rid, wage_past_month, where_poop___toilet, where_poop___river_lake, where_poop___forest, where_poop___latrine, where_poop___field ,electricity_resid, electricity_resid_source___power_lines, hh_item___tv, hh_item___smart_phone, highest_education_stage, vill_access_by___car, vill_access_by___truck, vill_access_by___boat, vill_access_by___none, vill_access_by___moto, house_type, water_source___filter, water_source___other, water_source___river_lake, water_source___well)

skimr::skim(vars_need)

vars_need$NA_count <- rowSums(is.na(vars_need))

hist(vars_need$NA_count)

columns_of_interest = colnames(vars_need)[2:22]

#df_filtered <- dat %>% filter(if_all(all_of(columns_of_interest), ~ !is.na(.x))) #1177
df_filtered <- dat %>% filter(rowSums(is.na(across(all_of(columns_of_interest)))) < 10) #1193

df_filtered$water_source___gov_tap <- 0
df_filtered[which(df_filtered$water_source___filter==0 & df_filtered$water_source___river_lake==0 & df_filtered$water_source___well==0),]$water_source___gov_tap <- 1

df_filtered <- df_filtered %>% 
  mutate(
    highest_where_poop = dplyr::case_when(
      where_poop___toilet == 1 ~ 1,
      where_poop___latrine == 1 ~ 0.5,
      where_poop___field == 1 ~ 0,
      where_poop___forest == 1 ~ 0,
      where_poop___river_lake == 1 ~ 0,
      TRUE ~ NA_real_),
    house_type_score = dplyr::case_when(
      house_type == "concrete" ~ 1,
      house_type == "wood" ~ 1,
      house_type == "traditional" ~ 0,
      TRUE ~ NA_real_),
    water_source_score = dplyr::case_when(
      water_source___filter == 1 ~ 1,
      water_source___well == 1 ~ 1,
      water_source___gov_tap == 1 ~ 1,
      water_source___river_lake == 1 ~ 0,
      TRUE ~ NA_real_))


##scaled 
df_filtered <- df_filtered %>% 
  rowwise() %>%
  mutate(h_sol = (sum(c(house_type_score, house_type_score,
                        water_source_score, highest_where_poop,
                        electricity_resid), na.rm = TRUE))/
           (sum(!is.na(c(house_type_score, house_type_score,
                         water_source_score, highest_where_poop,
                         electricity_resid))))) %>%
  ungroup()


##unscaled 
df_filtered <- df_filtered %>% 
    rowwise() %>%
    mutate(h_sol2 = (sum(c(house_type_score, house_type_score,
                          water_source_score, highest_where_poop,
                          electricity_resid), na.rm = TRUE))) %>% ungroup()

hist(df_filtered$h_sol)
hist(df_filtered$h_sol2)
```


add everything together 
```{r}
df_join = df_filtered %>% select(rid, h_sol, h_sol2)

df_join2 = left_join(dat, df_join, by="rid")

df_join2 %<>% dplyr::select(rid, h_sol, h_sol2, Early_urban_PC1, Early_urban_PC2)

df_join2 %<>% filter(rid %in% meta$rid)

#duplicated ids are just NAs
df_join2_unique <- df_join2 %>% arrange(rid, is.na(Early_urban_PC1)) %>% distinct(rid, .keep_all = TRUE)

dat2 = left_join(meta, df_join2_unique, by="rid")
```


add in updated urbanicity scores 
location 12 (Simoi Lama) needs to be given same score as location 2 (Dayok)
```{r}
urbanicity <- read.delim('~/vandy/early_life/redo_11nov25/OAHeLP_location_urbanicity_scores_2025-07-11.txt')
new_row <- data.frame(Date = "2022-06-13", village_id = 12, Name.of.village = "Simoi Lama", community_score=20.450, urb_score=24.620)
urbanicity2 <- rbind(urbanicity, new_row)
urbanicity2 %<>% dplyr::select(-Date)

names(dat2)[which(names(dat2) == "interview_location_med")] <- "village_id"
dat2 %<>% dplyr::select(-urb_score)

dat3 = left_join(dat2, urbanicity2, by="village_id")
```

removing samples with incomplete data 
```{r}
dim(subset(dat3,!is.na(age) & !is.na(sex_medical))) #984
dat4 = subset(dat3, age>=17.5) #967
dim(subset(dat4, !sex_medical == rnaseq_sex)) #3 keeping for now 
cat("final df size:", dim(dat4), "\n")
```

regress out batch effects
```{r}
dat4$age_scale<-as.numeric(scale(dat4$age))
dat4$urb_scale<-as.numeric(scale(dat4$urb_score))

v2=v[,dat4$sample_name]
identical(dat4$sample_name,colnames(v2))

design = model.matrix(~as.factor(dat4$batch) + dat4$Uniquely.mapped.reads + dat4$monocytes_perc_combine + dat4$lympho_perc_combine)
fit <-lmFit(v2,design)
fit <- eBayes(fit)
exp<-residuals.MArrayLM(object=fit, v2)
```


match with kinship data 
```{r}
OApbmcKinshipdata <- readRDS("~/vandy/early_life/redo_11nov25/OApbmcKinshipdata.rds")
colnames(OApbmcKinshipdata) <- paste0("X", colnames(OApbmcKinshipdata))
rownames(OApbmcKinshipdata) <- paste0("X", rownames(OApbmcKinshipdata))
dat4 %<>% filter(sample_name %in% colnames(OApbmcKinshipdata)) #963
K = OApbmcKinshipdata[dat4$sample_name,dat4$sample_name]
```


also adding in genomic pcs 
```{r}
eigenvecOApbmcs <- read_csv("~/vandy/early_life/redo_11nov25/eigenvecOApbmcs.csv")
eigenvecOApbmcs %<>% dplyr::select(sample_name, PC1, PC2)
colnames(eigenvecOApbmcs) <- c("sample_name", "genomic_PC1", "genomic_PC2")
dat4 = left_join(dat4, eigenvecOApbmcs, by="sample_name")
```


making sample map 
```{r}
sample_map = dat4 %>% dplyr::select(final_TID, rid, sample_name)
colnames(sample_map) = c("tid", "rid", "sample_name")
```


filter exp 
```{r}
exp2 = exp[,dat4$sample_name]
```


saving important stuff here!!!
```{r}
# saveRDS(exp2,file='~/vandy/early_life/redo_11nov25/exp_LARGE_25nov25_batch_effects_removed.rds')
```


### fixing groups ---


```{r}
rm(list = ls())
```

read in current meta data 
```{r}
meta <- readRDS("~/vandy/early_life/redo_11nov25/meta_data_13nov25_AFTER_group_fixes.rds")
```

remove other-ethnicicty and start over 
```{r}
personal=read.csv('~/vandy/early_life/redo_11nov25/personal_information.csv')
medical=read.csv('~/vandy/early_life/redo_11nov25//medical.csv')
```


from 25june25
```{r}
compile_ethnic <- function(personal_info = NULL, med = NULL){
  if(is.null(personal_info)) stop("ERROR: Must include valid dataframe as population register")
  
  # add medical visit village for contextualizing info
  personal_info$interview_location_med <- med$interview_location_med[match(personal_info$rid, med$rid)]
  personal_info$other_ethnicity[which(personal_info$birth_place_state_other == "BugIS INDONESIA")] <- "Bugis"
  
  # First, deal with "Other" category by standardizing to fewer number of categories
  personal_info$other_ethnicity <- tolower(personal_info$other_ethnicity)
  
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("bidayuh"))]<-'Bidayuh'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("bugis"))]<-'Bugis'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("dayak, orang melayu"))]<-'Dayak_Malay'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("dusun"))]<-'Dusun'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("iban"))]<-'Iban'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("father indo", "indo", "indonesia", "indonesian"))]<-'Indonesian'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("jah hut", "jahut"))]<-'Jahut'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("jahut dan cewang"))]<-'Jahut_Cewang'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("kentak", "kintaq"))]<-'Kintaq'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("lano"))]<-'Lanoh'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("mah meri"))]<-'MahMeri'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("melayu", "orang melayu"))]<-'Malay'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("murut"))]<-'Murut'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("kensiu"))]<-'Kensiu'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("kensiu, melayu (father)"))]<-'Kensiu_Malay'
  personal_info$other_ethnicity[which(personal_info$other_ethnicity %in% c("semelai"))]<-'Semelai'
  
  # special case of Jakun
  personal_info$ethnicity___2[which(personal_info$ethnicity___2 == 0 & personal_info$interview_location_med == 25)]<-1  # this is Kg Petoh, so almost certainly jakun
  
  # this case marked only temiar just needs to be moved
  personal_info$ethnicity___9[which(personal_info$other_ethnicity %in% c("temiar"))] <- 1
  personal_info$ethnicity___99[which(personal_info$other_ethnicity %in% c("temiar"))] <- 0
  
  ## Extract numbers associated with ethnicities 
  # Get columns that match the pattern "ethnicity___[number]"
  pattern_cols <- grep("ethnicity___\\d+", names(personal_info), value = TRUE)
  
  # Extract numbers from column names
  numbers <- as.numeric(sub(".*___(\\d+)", "\\1", pattern_cols))
  
  # Create list to store results
  result_list <- vector("list", nrow(personal_info))
  
  # For each row
  for (i in 1:nrow(personal_info)) {
    
    # Get indices where value is 1
    ones_indices <- which(personal_info[i, pattern_cols] == 1)
    
    # Store corresponding numbers in result list
    result_list[[i]] <- numbers[ones_indices]
  }
  
  ## Replace numbers with the ethnic group names
  result_list <- lapply(result_list, FUN=function(x){
    case_match(x,
               0 ~ "Batek",
               1 ~ "Jehai",
               2 ~ "Jakun",
               3 ~ "MahMeri",
               4 ~ "Mendriq",
               5 ~ "OrangKuala",
               6 ~ "OrangSeletar",
               7 ~ "Semai",
               8 ~ "SemaqBeri",
               9 ~ "Temiar",
               10 ~ "Temuan",
               99 ~ "Other")
  })
  
  ## Replace "Other" designations in list
  for(i in 1:length(result_list)){
    if(sum(result_list[[i]] == "Other") > 0){
      result_list[[i]] <- case_match(result_list[[i]], 
                                     "Other" ~ personal_info$other_ethnicity[[i]])
    }
    
    result_list[[i]][which(is.na(result_list[[i]]))] <- "Unknown"
  }
  
  
  
  ## Assign 'Multiple' to ethnicity if more than one ethnicity is indicated
  personal_info$number_ethnics <- unlist(lapply(result_list, length)) 
  
  ## Make unknown for anyone without ethnicity information
  result_list[which(personal_info$number_ethnics == 0)] <- "Unknown"
  
  
  ## Create a general "ethnicity" column for use
  personal_info$ethnicity <- unlist(lapply(result_list, paste, collapse="_"))
  
  # make broader groups
  personal_info$ethnicity_groups<-'Other/Mixed'
  personal_info$ethnicity_groups[which(personal_info$ethnicity %in% c('Batek','Jehai','Mendriq', "Kensiu", "Lanoh", "Kintaq"))]<-'Semang'
  personal_info$ethnicity_groups[which(personal_info$ethnicity %in% c('Temuan','Jakun','OrangSeletar','OrangKuala','Semelai'))]<-'Proto_Malay'
  personal_info$ethnicity_groups[which(personal_info$ethnicity %in% c('Temiar','Semai','SemaqBeri','MahMeri',"Jahut"))]<-'Senoi'
  
  return(personal_info)
}
```

apply to the data 
```{r}
group_fixes <- compile_ethnic(personal_info = personal, med = medical)
group_fixes %<>% dplyr::select(rid, ethnicity, other_ethnicity, ethnicity_groups)
```

remove old cols from meta data and add new ones
```{r}
meta %<>% select(-ethnicity, -ethnicity2, -other_ethnicity, -ethnicity_groups)
meta2 = left_join(meta, group_fixes, by="rid")

table(meta2$other_ethnicity, meta2$ethnicity_groups)
```

fix the last couple group errors 
```{r}
meta2$ethnicity_groups[which(meta2$other_ethnicity %in% c("Jahut_Cewang"))]<-'Senoi'
```

```{r}
table(meta2$other_ethnicity, meta2$ethnicity_groups)
```

```{r}
#saveRDS(meta2,file='~/vandy/early_life/redo_11nov25/meta_data_26nov25_AFTER_group_fixes.rds')
```



